## list of species


See Methods from raw files download for the sources.

Species abreviation| Binomial name                    | Source
-------------------|----------------------------------|-----------------------
ACAPL              | Acanthaster planci               | NCBI GCF_001949145.1
ACRMI              | Acropora millepora               | NCBI GCF_004143615.1
ADIVA              | Adineta vaga                     | ENSEMBL Release-45
AMPQU              | Amphimedon queenslandica         | ENSEMBL Release-45
ANOGA              | Anopheles gambiae                | ENSEMBL Release-45
AURAU              | Aurelia aurita                   | David Gold Google Drive Aurelia.Genome_v1.2        
BRALA              | Branchiostoma lanceolatum        | ENSEMBL Release-45
CAEEL              | Caenorhabditis elegans           | ENSEMBL Release-45              
CALMI              | Callorhinchus milii              | NCBI GCF_000165045.1
CAPOW              | Capitella teleta                 | ENSEMBL Release-45
CAPTE              | Capsaspora owczarzaki            | ENSEMBL Release-45
CHEMY              | Chelonia mydas                   | NCBI GCF_000344595.1
CIOIN              | Ciona intestinalis               | NCBI GCF_000224145.3
CLYHE              | Clytia hemisphaerica             | MARIMBA
CRAGI              | Crassostrea gigas                | ENSEMBL Release-45 
DANRE              | Danio rerio                      | NCBI GCF_000002035.6
DAPPU              | Daphnia pulex                    | ENSEMBL Release-45
DROME              | Drosophila melanogaster          | ENSEMBL Release-45
EUPSC              | Euprymna scolopes                | Lachesis assembly (Schmidbaur et al. in prep.)
EXAPA              | Exaiptasia pallida               | NCBI GCF_001417965.1
GALGA              | Gallus gallus                    | NCBI GCF_000002315.6
HELRO              | Helobdella robusta               | ENSEMBL Release-45
HIPCO              | Hippocampus comes                | NCBI GCF_001891065.1
HOFMI              | Hofstenia miamia                 | http://srivastavalab.rc.fas.harvard.edu
HOIHO              | Hoilungia hongkongensis          | https://bitbucket.org/molpalmuc/hoilungia-genome/src/master/tracks/
HOMSA              | Homo sapiens                     | NCBI GCF_000001405.39         
HYDVU              | Hydra vulgaris                   | NHGRI hydra2.0
IXOSC              | Ixodes scapularis                | ENSEMBL Release-45
LATCH              | Latimeria chalumnae              | NCBI GCF_000225785.1
LEPOC              | Lepisosteus oculatus             | NCBI GCF_000242695.1      
LINAN              | Lingula anatina                  | ENSEMBL Release-45
LOTGI              | Lottia gigantea                  | ENSEMBL Release-45
MAYZE              | Maylandia zebra                  | NCBI GCF_000238955.4      
MIZYE              | Mizuhopecten yessoensis          | Wang et al. 2017
MNELE              | Mnemiopsis leidyi                | NHGRI ML2.2
MUSMU              | Mus musculus                     | NCBI GCF_000001635.26
NEMVE              | Nematostella vectensis           | ENSEMBL Release-45
PARTE              | Parasteatoda tepidariorum        | NCBI GCF_000365465.2 
PLEBA              | Pleurobrachia bachei             | NCBI GCA_000695325.1, Neurobase      
PTYFL              | Ptychodera flava                 | OIST pfl_public_ver1.0
SACKO              | Saccoglossus kowalevskii         | OIST Sackov3
SALRO              | Salpingoeca rosetta              | ENSEMBL Release-45
SCHME              | Schmidtea mediterranea           | Planmine
STRMA              | Strigamia maritima               | ENSEMBL Release-45
STRPU              | Strongylocentrotus purpuratus    | ENSEMBL Release-45
SYCCI              | Sycon ciliatum                   | DataDryad
TRIAD              | Trichoplax adhaerens             | ENSEMBL Release-45     
TRICA              | Tribolium castaneum              | ENSEMBL Release-45
XENTR              | Xenopus tropicalis               | NCBI GCF_000004195.3

## External scripts
orthoFinderToOrthogroup.pl: https://github.com/nijibabulu/metazoan_synteny/tree/master/scripts

## 1. Download of the raw data
### 1.1. Download peptide files
42/49 peptide databases directly downloaded using curl (or copied from proj, or cloned with git clone):
ENSEMBL from release 45

```
#From NCBI: 16 genomes
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/405/GCF_000001405.39_GRCh38.p13/GCF_000001405.39_GRCh38.p13_protein.faa.gz -o HOMSA_NCBI_raw_pep.fa.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/635/GCF_000001635.26_GRCm38.p6/GCF_000001635.26_GRCm38.p6_protein.faa.gz -o MUSMU_NCBI_raw_pep.fa.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/035/GCF_000002035.6_GRCz11/GCF_000002035.6_GRCz11_protein.faa.gz -o DANRE_NCBI_raw_pep.fa.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/315/GCF_000002315.6_GRCg6a/GCF_000002315.6_GRCg6a_protein.faa.gz -o GALGA_NCBI_raw_pep.fa.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/004/195/GCF_000004195.3_Xenopus_tropicalis_v9.1/GCF_000004195.3_Xenopus_tropicalis_v9.1_protein.faa.gz -o XENTR_NCBI_raw_pep.fa.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/165/045/GCF_000165045.1_Callorhinchus_milii-6.1.3/GCF_000165045.1_Callorhinchus_milii-6.1.3_protein.faa.gz -o CALMI_NCBI_raw_pep.fa.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/004/143/615/GCF_004143615.1_amil_sf_1.1/GCF_004143615.1_amil_sf_1.1_protein.faa.gz -o ACRMI_NCBI_raw_pep.fa.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/224/145/GCF_000224145.3_KH/GCF_000224145.3_KH_protein.faa.gz -o CIOIN_NCBI_raw_pep.fa.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/225/785/GCF_000225785.1_LatCha1/GCF_000225785.1_LatCha1_protein.faa.gz -o LATCH_NCBI_raw_pep.fa.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/238/955/GCF_000238955.4_M_zebra_UMD2a/GCF_000238955.4_M_zebra_UMD2a_protein.faa.gz -o MAYZE_NCBI_raw_pep.fa.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/242/695/GCF_000242695.1_LepOcu1/GCF_000242695.1_LepOcu1_protein.faa.gz -o LEPOC_NCBI_raw_pep.fa.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/344/595/GCF_000344595.1_CheMyd_1.0/GCF_000344595.1_CheMyd_1.0_protein.faa.gz -o CHEMY_NCBI_raw_pep.fa.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/365/465/GCF_000365465.2_Ptep_2.0/GCF_000365465.2_Ptep_2.0_protein.faa.gz -o PARTE_NCBI_raw_pep.fa.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/001/417/965/GCF_001417965.1_Aiptasia_genome_1.1/GCF_001417965.1_Aiptasia_genome_1.1_protein.faa.gz -o EXAPA_NCBI_raw_pep.fa.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/001/891/065/GCF_001891065.1_H_comes_QL1_v1/GCF_001891065.1_H_comes_QL1_v1_protein.faa.gz -o HIPCO_NCBI_raw_pep.fa.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/001/949/145/GCF_001949145.1_OKI-Apl_1.0/GCF_001949145.1_OKI-Apl_1.0_protein.faa.gz -o ACAPL_NCBI_raw_pep.fa.gz

#From ENSEMBL: 20 genomes
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/fasta/adineta_vaga/pep/Adineta_vaga.AMS_PRJEB1171_v1.pep.all.fa.gz -o ADIVA_ENSEMBL_raw_pep.fa.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/fasta/amphimedon_queenslandica/pep/Amphimedon_queenslandica.Aqu1.pep.all.fa.gz -o AMPQU_ENSEMBL_raw_pep.fa.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/fasta/anopheles_gambiae/pep/Anopheles_gambiae.AgamP4.pep.all.fa.gz -o ANOGA_ENSEMBL_raw_pep.fa.gz
curl ftp://ftp.ensemblgenomes.org/pub/release-45/metazoa/fasta/branchiostoma_lanceolatum/pep/Branchiostoma_lanceolatum.BraLan2.pep.all.fa.gz -o BRALA_ENSEMBL_raw_pep.fa.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/fasta/caenorhabditis_elegans/pep/Caenorhabditis_elegans.WBcel235.pep.all.fa.gz -o CAEEL_ENSEMBL_raw_pep.fa.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/fasta/capitella_teleta/pep/Capitella_teleta.Capitella_teleta_v1.0.pep.all.fa.gz -o CAPTE_ENSEMBL_raw_pep.fa.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/fasta/crassostrea_gigas/pep/Crassostrea_gigas.oyster_v9.pep.all.fa.gz -o CRAGI_ENSEMBL_raw_pep.fa.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/fasta/daphnia_pulex/pep/Daphnia_pulex.V1.0.pep.all.fa.gz -o DAPPU_ENSEMBL_raw_pep.fa.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/fasta/drosophila_melanogaster/pep/Drosophila_melanogaster.BDGP6.22.pep.all.fa.gz -o DROME_ENSEMBL_raw_pep.fa.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/fasta/helobdella_robusta/pep/Helobdella_robusta.Helro1.pep.all.fa.gz -o HELRO_ENSEMBL_raw_pep.fa.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/fasta/ixodes_scapularis/pep/Ixodes_scapularis.IscaW1.pep.all.fa.gz -o IXOSC_ENSEMBL_raw_pep.fa.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/fasta/lingula_anatina/pep/Lingula_anatina.LinAna1.0.pep.all.fa.gz -o LINAN_ENSEMBL_raw_pep.fa.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/fasta/lottia_gigantea/pep/Lottia_gigantea.Lotgi1.pep.all.fa.gz -o LOTGI_ENSEMBL_raw_pep.fa.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/fasta/nematostella_vectensis/pep/Nematostella_vectensis.ASM20922v1.pep.all.fa.gz -o NEMVE_ENSEMBL_raw_pep.fa.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/fasta/strigamia_maritima/pep/Strigamia_maritima.Smar1.pep.all.fa.gz -o STRMA_ENSEMBL_raw_pep.fa.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/fasta/strongylocentrotus_purpuratus/pep/Strongylocentrotus_purpuratus.Spur_3.1.pep.all.fa.gz -o STRPU_ENSEMBL_raw_pep.fa.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/fasta/tribolium_castaneum/pep/Tribolium_castaneum.Tcas5.2.pep.all.fa.gz -o TRICA_ENSEMBL_raw_pep.fa.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/fasta/trichoplax_adhaerens/pep/Trichoplax_adhaerens.ASM15027v1.pep.all.fa.gz -o TRIAD_ENSEMBL_raw_pep.fa.gz
curl ftp://ftp.ensemblgenomes.org/pub/protists/release-45/fasta/protists_choanoflagellida1_collection/salpingoeca_rosetta_gca_000188695/pep/Salpingoeca_rosetta_gca_000188695.Proterospongia_sp_ATCC50818.pep.all.fa.gz -o SALRO_ENSEMBL_raw_pep.fa.gz
curl ftp://ftp.ensemblgenomes.org/pub/protists/release-45/fasta/protists_ichthyosporea1_collection/capsaspora_owczarzaki_atcc_30864_gca_000151315/pep/Capsaspora_owczarzaki_atcc_30864_gca_000151315.C_owczarzaki_V2.pep.all.fa.gz -o CAPOW_ENSEMBL_raw_pep.fa.gz

#From other sources: 7 genomes
curl http://marimba.obs-vlfr.fr/download/file/fid/50 -o CLYHE_OTHER_raw_pep.fa
curl https://research.nhgri.nih.gov/hydra/download/genemodels_proteins/hydra2.0_genemodels.aa.gz -o HYDVU_OTHER_raw_pep.fa.gz
git clone https://bitbucket.org/molpalmuc/hoilungia-genome/src/master/sequences/Hhon_BRAKER1_proteins.fasta.gz HOIHO_raw_pep.fa.gz
cp /proj/Simakov/OTHER/SCALLOP/PYgenome-pep.fa MIZYE_OTHER_raw_pep.fa
curl https://research.nhgri.nih.gov/mnemiopsis/download/proteome/ML2.2.aa.gz -o MNELE_OTHER_raw_pep.fa.gz
curl https://marinegenomics.oist.jp/acornworm/download/pfl_public_ver1.0.prot > PTYFL_OTHER_raw_pep.fa
```
The 7 remaining genomes were downloaded with a browser (and or built using annotation files with gffread):

* Aurelia aurita from: https://drive.google.com/drive/folders/1NC6bZ9cxWkZyofOsMPzrxIH3C7m1ySiu, Aurelia.Genome_v1.2_Protein_Models_12-28-18.fasta, renamed AURAU_raw_pep.fa
* Euprymna scolopes clusters peptide file was copied from: /proj/Simakov/EUPRYMNA/MAHDITRAN/Euprymna_scolopes.fa , named EUPSC_raw_pep.fa
* Hofstenia miamia gff3 file was downloaded from http://srivastavalab.rc.fas.harvard.edu gffread was used to build HOFMI_raw_pep.fa using hmi_gene_annotation.gff3 and hmi_genome.fa
* Saccoglossus kowalevskii: SkowalevskiiJGIv3.0.longestTrs.pep.fa.gz Downloaded from Metazome v3 (https://metazome.jgi.doe.gov/pz/portal.html#!bulk?org=Org_Skowalevskii_er), called SACKO_OTHER_raw_pep.fa.gz
* Schmidtea mediterranea: gffread used to build SCHME_OTHER_ra_pep.fa using gff http://planmine.mpi-cbg.de/planmine/model/bulkdata/smes_v2_hconf_SMESG.gff3.zip (high confidence transcripts) and  http://planmine.mpi-cbg.de/planmine/model/bulkdata/dd_Smes_g4.fasta.zip


```
gffread SCHME_OTHER.gff3 -g dd_Smes_g4.fasta -y SCHME_OTHER_raw_pep.fa
```
* Sycon ciliatum genome, CDS and peptide downloaded from datadryad (https://datadryad.org/resource/doi:10.5061/dryad.tn0f3).
* Pleurobrachia bachei filtered gene models mRNAs (CDS) were downloaded from https://neurobase.rc.ufl.edu/pleurobrachia/download (03_P-bachei_Filtered_Gene_Models_RNA.txt) and also translated into peptides (also using transeq, frame 1), file was named PLEBA_raw_pep.fa
	we use
	```sed -i 's/_1$//' PLEBA.fa``` to delete the frame added by transseq to the accession

Finally, we gunzip all the gz files 

```
gunzip *.fa.gz
```



### 1.2. Download annotation files
### 1.2.1 Download of available gff files
gff files were available for 47 out of the 49 species. For SYCCI and PLEBA, we'll map the transcripts (see 1.2.2)
39/49 downloaded using curl:
We want the gffs which show all the sequences. For example, in the case of a chromosomal level assembly, we want the unplaced scaffolds as well.

```
#16 Gffs downloaded from NCBI
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/001/949/145/GCF_001949145.1_OKI-Apl_1.0/GCF_001949145.1_OKI-Apl_1.0_genomic.gff.gz -o ACAPL_NCBI.gff3.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/004/143/615/GCF_004143615.1_amil_sf_1.1/GCF_004143615.1_amil_sf_1.1_genomic.gff.gz -o ACRMI_NCBI.gff.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/165/045/GCF_000165045.1_Callorhinchus_milii-6.1.3/GCF_000165045.1_Callorhinchus_milii-6.1.3_genomic.gff.gz -o CALMI_NCBI.gff3.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/224/145/GCF_000224145.3_KH/GCF_000224145.3_KH_genomic.gff.gz -o CIOIN_NCBI.gff3.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/035/GCF_000002035.6_GRCz11/GCF_000002035.6_GRCz11_genomic.gff.gz -o DANRE_NCBI.gff3.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/001/417/965/GCF_001417965.1_Aiptasia_genome_1.1/GCF_001417965.1_Aiptasia_genome_1.1_genomic.gff.gz -o EXAPA_NCBI.gff.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/315/GCF_000002315.6_GRCg6a/GCF_000002315.6_GRCg6a_genomic.gff.gz -o GALGA_NCBI.gff3.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/242/695/GCF_000242695.1_LepOcu1/GCF_000242695.1_LepOcu1_genomic.gff.gz -o LEPOC_NCBI.gff3.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/635/GCF_000001635.26_GRCm38.p6/GCF_000001635.26_GRCm38.p6_genomic.gff.gz -o MUSMU_NCBI.gff3.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/405/GCF_000001405.39_GRCh38.p13/GCF_000001405.39_GRCh38.p13_genomic.gff.gz -o HOMSA_NCBI.gff.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/365/465/GCF_000365465.2_Ptep_2.0/GCF_000365465.2_Ptep_2.0_genomic.gff.gz -o PARTE_NCBI.gff3.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/225/785/GCF_000225785.1_LatCha1/GCF_000225785.1_LatCha1_genomic.gff.gz -o LATCH_NCBI.gff.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/001/891/065/GCF_001891065.1_H_comes_QL1_v1/GCF_001891065.1_H_comes_QL1_v1_genomic.gff.gz -o HIPCO_NCBI.gff.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/238/955/GCF_000238955.4_M_zebra_UMD2a/GCF_000238955.4_M_zebra_UMD2a_genomic.gff.gz -o MAYZE_NCBI.gff.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/344/595/GCF_000344595.1_CheMyd_1.0/GCF_000344595.1_CheMyd_1.0_genomic.gff.gz -o CHEMY_NCBI.gff.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/004/195/GCF_000004195.3_Xenopus_tropicalis_v9.1/GCF_000004195.3_Xenopus_tropicalis_v9.1_genomic.gff.gz -o XENTR_NCBI.gff.gz

#20 Gffs downloaded from ENSEMBL
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/gff3/adineta_vaga/Adineta_vaga.AMS_PRJEB1171_v1.45.gff3.gz -o ADIVA_ENSEMBL.gff3.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/gff3/anopheles_gambiae//Anopheles_gambiae.AgamP4.45.gff3.gz -o ANOGA_ENSEMBL.gff3.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/gff3/caenorhabditis_elegans//Caenorhabditis_elegans.WBcel235.45.gff3.gz -o CAEEL_ENSEMBL.gff3.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/gff3/capitella_teleta//Capitella_teleta.Capitella_teleta_v1.0.45.gff3.gz -o CAPTE_ENSEMBL.gff3.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/gff3/crassostrea_gigas/Crassostrea_gigas.oyster_v9.45.gff3.gz -o CRAGI_ENSEMBL.gff3.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/gff3/daphnia_pulex/Daphnia_pulex.V1.0.45.gff3.gz -o DAPPU_ENSEMBL.gff3.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/gff3/drosophila_melanogaster/Drosophila_melanogaster.BDGP6.22.45.gff3.gz -o DROME_ENSEMBL.gff3.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/gff3/helobdella_robusta/Helobdella_robusta.Helro1.45.gff3.gz -o HELRO_ENSEMBL.gff3.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/gff3/ixodes_scapularis/Ixodes_scapularis.IscaW1.45.gff3.gz -o IXOSC_ENSEMBL.gff3.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/gff3/lingula_anatina/Lingula_anatina.LinAna1.0.45.gff3.gz -o LINAN_ENSEMBL.gff3.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/gff3/lottia_gigantea/Lottia_gigantea.Lotgi1.45.gff3.gz -o LOTGI_ENSEMBL.gff3.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/gff3/nematostella_vectensis//Nematostella_vectensis.ASM20922v1.45.gff3.gz -o NEMVE_ENSEMBL.gff3.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/gff3/strigamia_maritima//Strigamia_maritima.Smar1.45.gff3.gz -o STRMA_ENSEMBL.gff3.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/gff3/strongylocentrotus_purpuratus//Strongylocentrotus_purpuratus.Spur_3.1.45.gff3.gz -o STRPU_ENSEMBL.gff3.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/gff3/tribolium_castaneum/Tribolium_castaneum.Tcas5.2.45.gff3.gz -o TRICA_ENSEMBL.gff3.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/gff3/trichoplax_adhaerens//Trichoplax_adhaerens.ASM15027v1.45.gff3.gz -o TRIAD_ENSEMBL.gff3.gz
curl ftp://ftp.ensemblgenomes.org/pub/protists/release-45/gff3/protists_choanoflagellida1_collection/salpingoeca_rosetta_gca_000188695/Salpingoeca_rosetta_gca_000188695.Proterospongia_sp_ATCC50818.45.gff3.gz -o SALRO_ENSEMBL.gff3.gz
curl ftp://ftp.ensemblgenomes.org/pub/protists/release-45/gff3/protists_ichthyosporea1_collection/capsaspora_owczarzaki_atcc_30864_gca_000151315/Capsaspora_owczarzaki_atcc_30864_gca_000151315.C_owczarzaki_V2.45.gff3.gz -o CAPOW_ENSEMBL.gff3.gz
curl ftp://ftp.ensemblgenomes.org/pub/release-45/metazoa/gff3/amphimedon_queenslandica/Amphimedon_queenslandica.Aqu1.45.gff3.gz -o AMPQU_ENSEMBL.gff3.gz
curl ftp://ftp.ensemblgenomes.org/pub/release-45/metazoa/gff3/branchiostoma_lanceolatum/Branchiostoma_lanceolatum.BraLan2.45.gff3.gz -o BRALA_ENSEMBL.gff3.gz

#4 gffs downloaded from other ressources
curl https://research.nhgri.nih.gov/hydra/download/genemodels_gff3/hydra2.0_genemodels.gff3.gz -o HYDVU_OTHER.gff3.gz
curl curl https://bitbucket.org/molpalmuc/hoilungia-genome/raw/0d523a5b8556741a37918f3f30d0ed0414833912/tracks/Hhon_BRAKER1_CDS.gff3.gz -o HOIHO_OTHER.gff3.gz
curl https://marinegenomics.oist.jp/acornworm/download/pfl_public_ver1.0.gff3 > PTYFL_OTHER.gff3
curl https://research.nhgri.nih.gov/mnemiopsis/download/proteome/ML2.2.gff3.gz -o MNELE_OTHER.gff3.gz
```

* *Aurelia aurita* (https://drive.google.com/drive/folders/1NC6bZ9cxWkZyofOsMPzrxIH3C7m1ySiu), named AURAU_OTHER.gff3
* *Branchiostoma lanceolatum*, gtf downloaded from https://www.dropbox.com/s/d4fqnoa8gdix3pa/Bla_annot_final.gtf.gz named BRALA_OTHER.gtf.gz
* *Clytia hemisphaerica*, gff from http://marimba.obs-vlfr.fr/download/file/fid/51 called CLYHE_OTHER.gff3
* *Euprymna scolopes*: gff located in /proj/Simakov/EUPRYMNA/MAHDITRAN/clusters_esc.gff3 use gffparser in the next parts firectly on this file.
* *Hofstenia miamia*, gff3 downloaded from http://srivastavalab.rc.fas.harvard.edu, hmi_gene_annotation.gff3 was renamed HOFMI_OTHER.gff3
* *Hoilungia hongkongensis*, gff3 downloaded from https://bitbucket.org/molpalmuc/hoilungia-genome/src/master/tracks/Hhon_BRAKER1_CDS.gff3.gz named HOIHO_OTHER.gff3
* *Mizuhopecten yessoensis* gff located in /proj/Simakov/OTHER/SCALLOP/chr.id.gff3 use gffparser in the next parts directly on this file.
* *Schmidtea mediterranea*: donwloaded high confidence gene predictions from  http://planmine.mpi-cbg.de/planmine/aspect.do?name=Gene%20Predictions, called the file SCHME_OTHER.gff3
* *Saccoglossus kowalevskii*: downloaded the longest transcrfipts form Metazome v3    SkowalevskiiJGIv3.0.longestTrs.gff3.gz

### 1.2.2 Mapping of transcripts/gene models onto the genome when annotation files were not uploaded by the authors (Pleurobrachia)
The annotations for *Pleurobrachia bachei* gene models and *Sycon ciliatum* transcripts were not available.
Genome database build using **gmap_gbuild**, transcripts mapped onto the genome using **gmap**

```
module load gmap
gmap_build -D Pleurobrachia_bachei -d PLEBA_genome GCA_000695325.1_P.bachei_draft_genome_v.1.1_genomic.fna
gmap -D Pleurobrachia_bachei/ -d PLEBA_genome 03_P-bachei_Filtered_Gene_Models_RNA.txt --gff3-add-separators=1 -f gff3_gene > PLEBA_03_mapped.gff3 

gmap_build -D Sycon_ciliatum/ -d SYCCI_genome sycon.genome.fa
gmap -D Sycon_ciliatum/ -d SYCCI_genome sycon.cds.fa --gff3-add-separators=1 -f gff3_gene > SYCCI_mapped.gff3 
```

* *Pleurobrachia bachei* : **18871 out of 18950** transcripts are mapped (i.e. 99.58%)
* *Sycon ciliatum* : **49278 out of 50731** transcripts are mapped (i.e. 97.00%)

We'll keep all the peptides in the following analyses (and not only the ones mapped onto the genome), because we want to differentiate between loss of genes and loss of synteny.
Keep the best mapped transcript
```
grep mrna1 SYCCI_mapped.gff3 | grep -P '\texon\t' > SYCCI_mapped_final.gff3
sed -i 's/sctid/scpid/g' SYCCI_mapped_final.gff3
grep mrna1 PLEBA_03_mapped.gff3 | grep -P '\texon\t' > PLEBA_mapped_final.gff3
```
