## 2. Preparation of data for susbequent analyses
There is some preliminary steps we want to do before running the gff parser.

* For *H .miamia*, we modify the gff like this:

`sed -i '/\|/s/=.*\|/=/' HOFMI_hmi_gene_annotation.gff3`

* In the case of *S. ciliatum* and *P. bachei*, gmap outputs multiple paths per mRNA. We want to keep only one. Also, gmap maps both CDS and exons. Since we're mapping only CDS (some don't start with ATG), we'll keep only exons features.


```
grep mrna1 SYCCI_mapped.gff3 | grep -P '\texon\t' > SYCCI_mapped_final.gff3
sed -i 's/sctid/scpid/g' SYCCI_mapped_final.gff3
grep mrna1 PLEBA_03_mapped.gff3 | grep -P '\texon\t' > PLEBA_mapped_final.gff3
```

In the case of *S. kowalevskii* the accessions in the fasta files are gene names, and are not in the CDS lines. So we'll edit the gff with this script. It'll add up ";protein_id=Sakowv12345678m" comment in the last field of the gff on the lines containing CDS features. This is to make the file parseable by pyMakeMap

```
sk_gff_preparation.py SACKO_SkowalevskiiJGIv3.0.longestTrs.gff3 > SACKO_prepped.gff3
rm SACKO_SkowalevskiiJGIv3.0.longestTrs.gff3
```

### 2.1 Prepare non-redundant chrom and fasta files from gff
CHROM files are the format we commonly use. They're made by parsing gff annotation files. Some gffs comprise info about exon, gene, CDS, UTR.

We define gene coordinates as the CDS information for every species.

#### 2.1.1 Prepare chrom files for NCBI data
First, we'll delete the mitochondrial scaffolds from the gffs. Not all genome assemblies posess mitochondrial, we don't want mitochondrial cluster as falsely assigned synteny novelties.

```
grep "NC_007788.1" CALMI_NCBI.gff3| cut -f1 | grep -wvf - ACAPL_NCBI.gff3 > ACAPL_NCBI_filt.gff3; rm ACAPL_NCBI.gff3
#ACRMI has no MT chrom
grep "NC_014285.1" CALMI_NCBI.gff3| cut -f1 | grep -wvf - CALMI_NCBI.gff3 > CALMI_NCBI_filt.gff; rm CALMI_NCBI.gff3
grep "NC_017929.1" CIOIN_NCBI.gff3| cut -f1 | grep -wvf - CIOIN_NCBI.gff3 > CIOIN_NCBI_filt.gff3; rm CIOIN_NCBI.gff3
grep "NC_002333.2" DANRE_NCBI.gff3| cut -f1 | grep -wvf - DANRE_NCBI.gff3 > DANRE_filt_NCBI.gff3; rm DANRE_NCBI.gff3
#EXAPA has no MT chrom
grep -v "NC_040902.1" GALGA_NCBI.gff3 > GALGA_filt_NCBI.gff3; rm GALGA_NCBI.gff3
grep "NC_004744.1" LEPOC_NCBI.gff3| cut -f1 | grep -wvf - LEPOC_NCBI.gff3 > LEPOC_filt_NCBI.gff3; rm LEPOC_NCBI.gff3
grep "NC_005089.1" MUSMU_NCBI.gff3| cut -f1 | grep -wvf - MUSMU_NCBI.gff3 > MUSMU_filt_NCBI.gff3; rm MUSMU_NCBI.gff3
grep "NC_012920.1" HOMSA_NCBI.gff| cut -f1|   grep -wvf - HOMSA_NCBI.gff  > HOMSA_filt_NCBI.gff; rm HOMSA_NCBI.gff
#PARTE has no MT
grep "NC_001804.1" LATCH_NCBI.gff|cut -f1|grep -wvf - LATCH_NCBI.gff > LATCH_filt_NCBI.gff; rm LATCH_NCBI.gff
grep "NC_020336.1" HIPCO_NCBI.gff|cut -f1|grep -wvf - HIPCO_NCBI.gff > HIPCO_filt_NCBI.gff; rm HIPCO_NCBI.gff
grep "NC_027944.1" MAYZE_NCBI.gff|cut -f1|grep -wvf - MAYZE_NCBI.gff > MAYZE_filt_NCBI.gff; rm MAYZE_NCBI.gff
grep "NC_000886.1" CHEMY_NCBI.gff|cut -f1|grep -wvf - CHEMY_NCBI.gff > CHEMY_NCBI_filt.gff; rm CHEMY_NCBI.gff
grep "NC_006839.1" XENTR_NCBI.gff|cut -f1|grep -wvf - XENTR_NCBI.gff > XENTR_filt_NCBI.gff; rm XENTR_NCBI.gff
```

We use custom scripts to filters the longest isoforms (as identified by shared gene IDs). ENSEMBL gene ids are in the fasta headers. In the case of NCBI, gene IDs are not in the fasta file, so the annotation file (gff format) is also needed to determine to which gene the isoforms come from.

* ACAPL

```
LongestIsoforms_NCBI.py ../../proteins/ACAPL_NCBI_raw_pep.fa ACAPL_NCBI_filt.gff3 ACAPL_NCBI_longest_isoforms.fa
sed -i 's/^>/>ACAPL_/' ACAPL_NCBI_longest_isoforms.fa
pymakeMap.py -gff ACAPL_NCBI_filt.gff3 -p ACAPL -f CDS -k protein_id -o ../chrom/ACAPL.chrom -r fasta -F ACAPL_NCBI_longest_isoforms.fa
cut -f2 ../chrom/ACAPL.chrom > ACAPL.list ; pyfasta extract --file ACAPL.list --header --space --fasta ACAPL_NCBI_longest_isoforms.fa > ../../proteins_processed/ACAPL.fasta
rm ACAPL_NCBI_longest_isoforms* ACAPL.list
```

* ACRMI

```
LongestIsoforms_NCBI.py ../../proteins/ACRMI_NCBI_raw_pep.fa ACRMI_NCBI.gff ACRMI_NCBI_longest_isoforms.fa
sed -i 's/^>/>ACRMI_/' ACRMI_NCBI_longest_isoforms.fa
pymakeMap.py -gff ACRMI_NCBI.gff -p ACRMI -f CDS -k protein_id -o ../chrom/ACRMI.chrom -r fasta -F ACRMI_NCBI_longest_isoforms.fa
cut -f2 ../chrom/ACRMI.chrom > ACRMI.list ; pyfasta extract --file ACRMI.list --header --space --fasta ACRMI_NCBI_longest_isoforms.fa > ../../proteins_processed/ACRMI.fasta
rm ACRMI_NCBI_longest_isoforms* ACRMI.list
```

* CALMI

```
LongestIsoforms_NCBI.py ../../proteins/CALMI_NCBI_raw_pep.fa CALMI_NCBI.gff3 CALMI_NCBI_longest_isoforms.fa
sed -i 's/^>/>CALMI_/' CALMI_NCBI_longest_isoforms.fa
pymakeMap.py -gff CALMI_NCBI.gff3 -p CALMI -f CDS -k protein_id -o ../chrom/CALMI.chrom -r fasta -F CALMI_NCBI_longest_isoforms.fa
cut -f2 ../chrom/CALMI.chrom > CALMI.list ; pyfasta extract --file CALMI.list --header --space --fasta CALMI_NCBI_longest_isoforms.fa > ../../proteins_processed/CALMI.fasta
rm CALMI_NCBI_longest_isoforms* CALMI.list
```

* CHEMY

```
LongestIsoforms_NCBI.py ../../proteins/CHEMY_NCBI_raw_pep.fa CHEMY_NCBI_filt.gff CHEMY_NCBI_longest_isoforms.fa
sed -i 's/^>/>CHEMY_/' CHEMY_NCBI_longest_isoforms.fa
pymakeMap.py -gff CHEMY_NCBI_filt.gff -p CHEMY -f CDS -k protein_id -o ../chrom/CHEMY.chrom -r fasta -F CHEMY_NCBI_longest_isoforms.fa
cut -f2 ../chrom/CHEMY.chrom > CHEMY.list ; pyfasta extract --file CHEMY.list --header --space --fasta CHEMY_NCBI_longest_isoforms.fa > ../../proteins_processed/CHEMY.fasta
rm CHEMY_NCBI_longest_isoforms* CHEMY.list
```

* CIOIN

```
LongestIsoforms_NCBI.py ../../proteins/CIOIN_NCBI_raw_pep.fa CIOIN_NCBI_filt.gff3 CIOIN_NCBI_longest_isoforms.fa
sed -i 's/^>/>CIOIN_/' CIOIN_NCBI_longest_isoforms.fa
pymakeMap.py -gff CIOIN_NCBI_filt.gff3 -p CIOIN -f CDS -k protein_id -o ../chrom/CIOIN.chrom -r fasta -F CIOIN_NCBI_longest_isoforms.fa
cut -f2 ../chrom/CIOIN.chrom > CIOIN.list ; pyfasta extract --file CIOIN.list --header --space --fasta CIOIN_NCBI_longest_isoforms.fa > ../../proteins_processed/CIOIN.fasta
rm CIOIN_NCBI_longest_isoforms* CIOIN.list
```

* DANRE

```
LongestIsoforms_NCBI.py ../../proteins/DANRE_NCBI_raw_pep.fa DANRE_filt_NCBI.gff3 DANRE_NCBI_longest_isoforms.fa
sed -i 's/^>/>DANRE_/' DANRE_NCBI_longest_isoforms.fa
pymakeMap.py -gff DANRE_filt_NCBI.gff3 -p DANRE -f CDS -k protein_id -o ../chrom/DANRE.chrom -r fasta -F DANRE_NCBI_longest_isoforms.fa
cut -f2 ../chrom/DANRE.chrom > DANRE.list ; pyfasta extract --file DANRE.list --header --space --fasta DANRE_NCBI_longest_isoforms.fa > ../../proteins_processed/DANRE.fasta
rm DANRE_NCBI_longest_isoforms* DANRE.list
```

* EXAPA

```
LongestIsoforms_NCBI.py ../../proteins/EXAPA_NCBI_raw_pep.fa EXAPA_NCBI.gff EXAPA_NCBI_longest_isoforms.fa
sed -i 's/^>/>EXAPA_/' EXAPA_NCBI_longest_isoforms.fa
pymakeMap.py -gff EXAPA_NCBI.gff -p EXAPA -f CDS -k protein_id -o ../chrom/EXAPA.chrom -r fasta -F EXAPA_NCBI_longest_isoforms.fa
cut -f2 ../chrom/EXAPA.chrom > EXAPA.list ; pyfasta extract --file EXAPA.list --header --space --fasta EXAPA_NCBI_longest_isoforms.fa > ../../proteins_processed/EXAPA.fasta
rm EXAPA_NCBI_longest_isoforms* EXAPA.list
```

* GALGA

```
LongestIsoforms_NCBI.py ../../proteins/GALGA_NCBI_raw_pep.fa GALGA_filt_NCBI.gff3 GALGA_NCBI_longest_isoforms.fa
sed -i 's/^>/>GALGA_/' GALGA_NCBI_longest_isoforms.fa
pymakeMap.py -gff GALGA_filt_NCBI.gff3 -p GALGA -f CDS -k protein_id -o ../chrom/GALGA.chrom -r fasta -F GALGA_NCBI_longest_isoforms.fa
cut -f2 ../chrom/GALGA.chrom > GALGA.list ; pyfasta extract --file GALGA.list --header --space --fasta GALGA_NCBI_longest_isoforms.fa > ../../proteins_processed/GALGA.fasta
rm GALGA_NCBI_longest_isoforms* GALGA.list
```

* HIPCO

```
LongestIsoforms_NCBI.py ../../proteins/HIPCO_NCBI_raw_pep.fa HIPCO_filt_NCBI.gff HIPCO_NCBI_longest_isoforms.fa
sed -i 's/^>/>HIPCO_/' HIPCO_NCBI_longest_isoforms.fa
pymakeMap.py -gff HIPCO_filt_NCBI.gff -p HIPCO -f CDS -k protein_id -o ../chrom/HIPCO.chrom -r fasta -F HIPCO_NCBI_longest_isoforms.fa
cut -f2 ../chrom/HIPCO.chrom > HIPCO.list ; pyfasta extract --file HIPCO.list --header --space --fasta HIPCO_NCBI_longest_isoforms.fa > ../../proteins_processed/HIPCO.fasta
rm HIPCO_NCBI_longest_isoforms* HIPCO.list
```

* HOMSA

```
LongestIsoforms_NCBI.py ../../proteins/HOMSA_NCBI_raw_pep.fa HOMSA_filt_NCBI.gff HOMSA_NCBI_longest_isoforms.fa
sed -i 's/^>/>HOMSA_/' HOMSA_NCBI_longest_isoforms.fa
pymakeMap.py -gff HOMSA_filt_NCBI.gff -p HOMSA -f CDS -k protein_id -o ../chrom/HOMSA.chrom -r fasta -F HOMSA_NCBI_longest_isoforms.fa
cut -f2 ../chrom/HOMSA.chrom > HOMSA.list ; pyfasta extract --file HOMSA.list --header --space --fasta HOMSA_NCBI_longest_isoforms.fa > ../../proteins_processed/HOMSA.fasta
rm HOMSA_NCBI_longest_isoforms* HOMSA.list
```

* LATCH

```
LongestIsoforms_NCBI.py ../../proteins/LATCH_NCBI_raw_pep.fa LATCH_filt_NCBI.gff LATCH_NCBI_longest_isoforms.fa
sed -i 's/^>/>LATCH_/' LATCH_NCBI_longest_isoforms.fa
pymakeMap.py -gff LATCH_filt_NCBI.gff -p LATCH -f CDS -k protein_id -o ../chrom/LATCH.chrom -r fasta -F LATCH_NCBI_longest_isoforms.fa
cut -f2 ../chrom/LATCH.chrom > LATCH.list ; pyfasta extract --file LATCH.list --header --space --fasta LATCH_NCBI_longest_isoforms.fa > ../../proteins_processed/LATCH.fasta
rm LATCH_NCBI_longest_isoforms* LATCH.list
```

* LEPOC

```
LongestIsoforms_NCBI.py ../../proteins/LEPOC_NCBI_raw_pep.fa LEPOC_filt_NCBI.gff3 LEPOC_NCBI_longest_isoforms.fa
sed -i 's/^>/>LEPOC_/' LEPOC_NCBI_longest_isoforms.fa
pymakeMap.py -gff LEPOC_filt_NCBI.gff3 -p LEPOC -f CDS -k protein_id -o ../chrom/LEPOC.chrom -r fasta -F LEPOC_NCBI_longest_isoforms.fa
cut -f2 ../chrom/LEPOC.chrom > LEPOC.list ; pyfasta extract --file LEPOC.list --header --space --fasta LEPOC_NCBI_longest_isoforms.fa > ../../proteins_processed/LEPOC.fasta
rm LEPOC_NCBI_longest_isoforms* LEPOC.list
```

* MAYZE

```
LongestIsoforms_NCBI.py ../../proteins/MAYZE_NCBI_raw_pep.fa MAYZE_filt_NCBI.gff MAYZE_NCBI_longest_isoforms.fa
sed -i 's/^>/>MAYZE_/' MAYZE_NCBI_longest_isoforms.fa
pymakeMap.py -gff MAYZE_filt_NCBI.gff -p MAYZE -f CDS -k protein_id -o ../chrom/MAYZE.chrom -r fasta -F MAYZE_NCBI_longest_isoforms.fa
cut -f2 ../chrom/MAYZE.chrom > MAYZE.list ; pyfasta extract --file MAYZE.list --header --space --fasta MAYZE_NCBI_longest_isoforms.fa > ../../proteins_processed/MAYZE.fasta
rm MAYZE_NCBI_longest_isoforms* MAYZE.list
```

* MUSMU

```
LongestIsoforms_NCBI.py ../../proteins/MUSMU_NCBI_raw_pep.fa MUSMU_filt_NCBI.gff3 MUSMU_NCBI_longest_isoforms.fa
sed -i 's/^>/>MUSMU_/' MUSMU_NCBI_longest_isoforms.fa
pymakeMap.py -gff MUSMU_filt_NCBI.gff3 -p MUSMU -f CDS -k protein_id -o ../chrom/MUSMU.chrom -r fasta -F MUSMU_NCBI_longest_isoforms.fa
cut -f2 ../chrom/MUSMU.chrom > MUSMU.list ; pyfasta extract --file MUSMU.list --header --space --fasta MUSMU_NCBI_longest_isoforms.fa > ../../proteins_processed/MUSMU.fasta
rm MUSMU_NCBI_longest_isoforms* MUSMU.list
```

* PARTE

```
LongestIsoforms_NCBI.py ../../proteins/PARTE_NCBI_raw_pep.fa PARTE_NCBI.gff3 PARTE_NCBI_longest_isoforms.fa
sed -i 's/^>/>PARTE_/' PARTE_NCBI_longest_isoforms.fa
pymakeMap.py -gff PARTE_NCBI.gff3 -p PARTE -f CDS -k protein_id -o ../chrom/PARTE.chrom -r fasta -F PARTE_NCBI_longest_isoforms.fa
cut -f2 ../chrom/PARTE.chrom > PARTE.list ; pyfasta extract --file PARTE.list --header --space --fasta PARTE_NCBI_longest_isoforms.fa > ../../proteins_processed/PARTE.fasta
rm PARTE_NCBI_longest_isoforms* PARTE.list
```

* XENTR

```
LongestIsoforms_NCBI.py ../../proteins/XENTR_NCBI_raw_pep.fa XENTR_filt_NCBI.gff XENTR_NCBI_longest_isoforms.fa
sed -i 's/^>/>XENTR_/' XENTR_NCBI_longest_isoforms.fa
pymakeMap.py -gff XENTR_filt_NCBI.gff -p XENTR -f CDS -k protein_id -o ../chrom/XENTR.chrom -r fasta -F XENTR_NCBI_longest_isoforms.fa
cut -f2 ../chrom/XENTR.chrom > XENTR.list ; pyfasta extract --file XENTR.list --header --space --fasta XENTR_NCBI_longest_isoforms.fa > ../../proteins_processed/XENTR.fasta
rm XENTR_NCBI_longest_isoforms* XENTR.list
```

#### 2.1.2 Prepare chrom files for ENSEMBL data

* ADIVA

```
LongestIsoforms_ENSEMBL.py ../../proteins/ADIVA_ENSEMBL_raw_pep.fa ADIVA_ENSEMBL_longest_isoforms.fa
sed -i 's/^>/>ADIVA_/' ADIVA_ENSEMBL_longest_isoforms.fa
pymakeMap.py -gff ADIVA_ENSEMBL.gff3 -p ADIVA -f CDS -k protein_id -o ../chrom/ADIVA.chrom -r fasta -F ADIVA_ENSEMBL_longest_isoforms.fa
cut -f2 ../chrom/ADIVA.chrom > ADIVA.list ; pyfasta extract --file ADIVA.list --header --space --fasta ADIVA_ENSEMBL_longest_isoforms.fa > ../../proteins_processed/ADIVA.fasta
rm ADIVA_ENSEMBL_longest_isoforms* ADIVA.list
```

* AMPQU

```
LongestIsoforms_ENSEMBL.py ../../proteins/AMPQU_ENSEMBL_raw_pep.fa AMPQU_ENSEMBL_longest_isoforms.fa
sed -i 's/^>/>AMPQU_/' AMPQU_ENSEMBL_longest_isoforms.fa
pymakeMap.py -gff AMPQU_ENSEMBL.gff3 -p AMPQU -f CDS -k protein_id -o ../chrom/AMPQU.chrom -r fasta -F AMPQU_ENSEMBL_longest_isoforms.fa
cut -f2 ../chrom/AMPQU.chrom > AMPQU.list ; pyfasta extract --file AMPQU.list --header --space --fasta AMPQU_ENSEMBL_longest_isoforms.fa > ../../proteins_processed/AMPQU.fasta
rm AMPQU_ENSEMBL_longest_isoforms* AMPQU.list
```

* ANOGA

```
grep -vP "Mt\tVectorBase"  ANOGA_ENSEMBL.gff3 > ANOGA_filt_ENSEMBL.gff3; rm ANOGA_ENSEMBL.gff3
LongestIsoforms_ENSEMBL.py ../../proteins/ANOGA_ENSEMBL_raw_pep.fa ANOGA_ENSEMBL_longest_isoforms.fa
sed -i 's/^>/>ANOGA_/' ANOGA_ENSEMBL_longest_isoforms.fa
pymakeMap.py -gff ANOGA_filt_ENSEMBL.gff3 -p ANOGA -f CDS -k protein_id -o ../chrom/ANOGA.chrom -r fasta -F ANOGA_ENSEMBL_longest_isoforms.fa
cut -f2 ../chrom/ANOGA.chrom > ANOGA.list ; pyfasta extract --file ANOGA.list --header --space --fasta ANOGA_ENSEMBL_longest_isoforms.fa > ../../proteins_processed/ANOGA.fasta
rm ANOGA_ENSEMBL_longest_isoforms* ANOGA.list
```

* BRALA

```
LongestIsoforms_ENSEMBL.py ../../proteins/BRALA_ENSEMBL_raw_pep.fa BRALA_ENSEMBL_longest_isoforms.fa
sed -i 's/^>/>BRALA_/' BRALA_ENSEMBL_longest_isoforms.fa
pymakeMap.py -gff BRALA_ENSEMBL.gff3 -p BRALA -f CDS -k protein_id -o ../chrom/BRALA.chrom -r fasta -F BRALA_ENSEMBL_longest_isoforms.fa
cut -f2 ../chrom/BRALA.chrom > BRALA.list ; pyfasta extract --file BRALA.list --header --space --fasta BRALA_ENSEMBL_longest_isoforms.fa > ../../proteins_processed/BRALA.fasta
rm BRALA_ENSEMBL_longest_isoforms* BRALA.list
```

* CAEEL

```
grep -vP "MtDNA\tWormBase"  CAEEL_ENSEMBL.gff3 > CAEEL_filt_ENSEMBL.gff3; rm CAEEL_ENSEMBL.gff3
LongestIsoforms_ENSEMBL.py ../../proteins/CAEEL_ENSEMBL_raw_pep.fa CAEEL_ENSEMBL_longest_isoforms.fa
sed -i 's/^>/>CAEEL_/' CAEEL_ENSEMBL_longest_isoforms.fa
pymakeMap.py -gff CAEEL_filt_ENSEMBL.gff3 -p CAEEL -f CDS -k protein_id -o ../chrom/CAEEL.chrom -r fasta -F CAEEL_ENSEMBL_longest_isoforms.fa
cut -f2 ../chrom/CAEEL.chrom > CAEEL.list ; pyfasta extract --file CAEEL.list --header --space --fasta CAEEL_ENSEMBL_longest_isoforms.fa > ../../proteins_processed/CAEEL.fasta
rm CAEEL_ENSEMBL_longest_isoforms* CAEEL.list
```

* CAPOW

```
LongestIsoforms_ENSEMBL.py ../../proteins/CAPOW_ENSEMBL_raw_pep.fa CAPOW_ENSEMBL_longest_isoforms.fa
sed -i 's/^>/>CAPOW_/' CAPOW_ENSEMBL_longest_isoforms.fa
pymakeMap.py -gff CAPOW_ENSEMBL.gff3 -p CAPOW -f CDS -k protein_id -o ../chrom/CAPOW.chrom -r fasta -F CAPOW_ENSEMBL_longest_isoforms.fa
cut -f2 ../chrom/CAPOW.chrom > CAPOW.list ; pyfasta extract --file CAPOW.list --header --space --fasta CAPOW_ENSEMBL_longest_isoforms.fa > ../../proteins_processed/CAPOW.fasta
rm CAPOW_ENSEMBL_longest_isoforms* CAPOW.list
```

* CAPTE

```
LongestIsoforms_ENSEMBL.py ../../proteins/CAPTE_ENSEMBL_raw_pep.fa CAPTE_ENSEMBL_longest_isoforms.fa
sed -i 's/^>/>CAPTE_/' CAPTE_ENSEMBL_longest_isoforms.fa
pymakeMap.py -gff CAPTE_ENSEMBL.gff3 -p CAPTE -f CDS -k protein_id -o ../chrom/CAPTE.chrom -r fasta -F CAPTE_ENSEMBL_longest_isoforms.fa
cut -f2 ../chrom/CAPTE.chrom > CAPTE.list ; pyfasta extract --file CAPTE.list --header --space --fasta CAPTE_ENSEMBL_longest_isoforms.fa > ../../proteins_processed/CAPTE.fasta
rm CAPTE_ENSEMBL_longest_isoforms* CAPTE.list
```

* CRAGI

```
LongestIsoforms_ENSEMBL.py ../../proteins/CRAGI_ENSEMBL_raw_pep.fa CRAGI_ENSEMBL_longest_isoforms.fa
sed -i 's/^>/>CRAGI_/' CRAGI_ENSEMBL_longest_isoforms.fa
pymakeMap.py -gff CRAGI_ENSEMBL.gff3 -p CRAGI -f CDS -k protein_id -o ../chrom/CRAGI.chrom -r fasta -F CRAGI_ENSEMBL_longest_isoforms.fa
cut -f2 ../chrom/CRAGI.chrom > CRAGI.list ; pyfasta extract --file CRAGI.list --header --space --fasta CRAGI_ENSEMBL_longest_isoforms.fa > ../../proteins_processed/CRAGI.fasta
rm CRAGI_ENSEMBL_longest_isoforms* CRAGI.list
```

* DAPPU

```
LongestIsoforms_ENSEMBL.py ../../proteins/DAPPU_ENSEMBL_raw_pep.fa DAPPU_ENSEMBL_longest_isoforms.fa
sed -i 's/^>/>DAPPU_/' DAPPU_ENSEMBL_longest_isoforms.fa
pymakeMap.py -gff DAPPU_ENSEMBL.gff3 -p DAPPU -f CDS -k protein_id -o ../chrom/DAPPU.chrom -r fasta -F DAPPU_ENSEMBL_longest_isoforms.fa
cut -f2 ../chrom/DAPPU.chrom > DAPPU.list ; pyfasta extract --file DAPPU.list --header --space --fasta DAPPU_ENSEMBL_longest_isoforms.fa > ../../proteins_processed/DAPPU.fasta
rm DAPPU_ENSEMBL_longest_isoforms* DAPPU.list
```

* DROME

```
grep -vP "mitochondrion_genome\tFlyBase"  DROME_ENSEMBL.gff3 > DROME_filt_ENSEMBL.gff3; rm DROME_ENSEMBL.gff3
LongestIsoforms_ENSEMBL.py ../../proteins/DROME_ENSEMBL_raw_pep.fa DROME_ENSEMBL_longest_isoforms.fa
sed -i 's/^>/>DROME_/' DROME_ENSEMBL_longest_isoforms.fa
pymakeMap.py -gff DROME_filt_ENSEMBL.gff3 -p DROME -f CDS -k protein_id -o ../chrom/DROME.chrom -r fasta -F DROME_ENSEMBL_longest_isoforms.fa
cut -f2 ../chrom/DROME.chrom > DROME.list ; pyfasta extract --file DROME.list --header --space --fasta DROME_ENSEMBL_longest_isoforms.fa > ../../proteins_processed/DROME.fasta
rm DROME_ENSEMBL_longest_isoforms* DROME.list
```

* HELRO

```
LongestIsoforms_ENSEMBL.py ../../proteins/HELRO_ENSEMBL_raw_pep.fa HELRO_ENSEMBL_longest_isoforms.fa
sed -i 's/^>/>HELRO_/' HELRO_ENSEMBL_longest_isoforms.fa
pymakeMap.py -gff HELRO_ENSEMBL.gff3 -p HELRO -f CDS -k protein_id -o ../chrom/HELRO.chrom -r fasta -F HELRO_ENSEMBL_longest_isoforms.fa
cut -f2 ../chrom/HELRO.chrom > HELRO.list ; pyfasta extract --file HELRO.list --header --space --fasta HELRO_ENSEMBL_longest_isoforms.fa > ../../proteins_processed/HELRO.fasta
rm HELRO_ENSEMBL_longest_isoforms* HELRO.list
```

* IXOSC

```
LongestIsoforms_ENSEMBL.py ../../proteins/IXOSC_ENSEMBL_raw_pep.fa IXOSC_ENSEMBL_longest_isoforms.fa
sed -i 's/^>/>IXOSC_/' IXOSC_ENSEMBL_longest_isoforms.fa
pymakeMap.py -gff IXOSC_ENSEMBL.gff3 -p IXOSC -f CDS -k protein_id -o ../chrom/IXOSC.chrom -r fasta -F IXOSC_ENSEMBL_longest_isoforms.fa
cut -f2 ../chrom/IXOSC.chrom > IXOSC.list ; pyfasta extract --file IXOSC.list --header --space --fasta IXOSC_ENSEMBL_longest_isoforms.fa > ../../proteins_processed/IXOSC.fasta
rm IXOSC_ENSEMBL_longest_isoforms* IXOSC.list
```

* LINAN

```
LongestIsoforms_ENSEMBL.py ../../proteins/LINAN_ENSEMBL_raw_pep.fa LINAN_ENSEMBL_longest_isoforms.fa
sed -i 's/^>/>LINAN_/' LINAN_ENSEMBL_longest_isoforms.fa
pymakeMap.py -gff LINAN_ENSEMBL.gff3 -p LINAN -f CDS -k protein_id -o ../chrom/LINAN.chrom -r fasta -F LINAN_ENSEMBL_longest_isoforms.fa
cut -f2 ../chrom/LINAN.chrom > LINAN.list ; pyfasta extract --file LINAN.list --header --space --fasta LINAN_ENSEMBL_longest_isoforms.fa > ../../proteins_processed/LINAN.fasta
rm LINAN_ENSEMBL_longest_isoforms* LINAN.list
```

* LOTGI

```
LongestIsoforms_ENSEMBL.py ../../proteins/LOTGI_ENSEMBL_raw_pep.fa LOTGI_ENSEMBL_longest_isoforms.fa
sed -i 's/^>/>LOTGI_/' LOTGI_ENSEMBL_longest_isoforms.fa
pymakeMap.py -gff LOTGI_ENSEMBL.gff3 -p LOTGI -f CDS -k protein_id -o ../chrom/LOTGI.chrom -r fasta -F LOTGI_ENSEMBL_longest_isoforms.fa
cut -f2 ../chrom/LOTGI.chrom > LOTGI.list ; pyfasta extract --file LOTGI.list --header --space --fasta LOTGI_ENSEMBL_longest_isoforms.fa > ../../proteins_processed/LOTGI.fasta
rm LOTGI_ENSEMBL_longest_isoforms* LOTGI.list
```

* NEMVE

```
LongestIsoforms_ENSEMBL.py ../../proteins/NEMVE_ENSEMBL_raw_pep.fa NEMVE_ENSEMBL_longest_isoforms.fa
sed -i 's/^>/>NEMVE_/' NEMVE_ENSEMBL_longest_isoforms.fa
pymakeMap.py -gff NEMVE_ENSEMBL.gff3 -p NEMVE -f CDS -k protein_id -o ../chrom/NEMVE.chrom -r fasta -F NEMVE_ENSEMBL_longest_isoforms.fa
cut -f2 ../chrom/NEMVE.chrom > NEMVE.list ; pyfasta extract --file NEMVE.list --header --space --fasta NEMVE_ENSEMBL_longest_isoforms.fa > ../../proteins_processed/NEMVE.fasta
rm NEMVE_ENSEMBL_longest_isoforms* NEMVE.list
```

* SALRO

```
LongestIsoforms_ENSEMBL.py ../../proteins/SALRO_ENSEMBL_raw_pep.fa SALRO_ENSEMBL_longest_isoforms.fa
sed -i 's/^>/>SALRO_/' SALRO_ENSEMBL_longest_isoforms.fa
pymakeMap.py -gff SALRO_ENSEMBL.gff3 -p SALRO -f CDS -k protein_id -o ../chrom/SALRO.chrom -r fasta -F SALRO_ENSEMBL_longest_isoforms.fa
cut -f2 ../chrom/SALRO.chrom > SALRO.list ; pyfasta extract --file SALRO.list --header --space --fasta SALRO_ENSEMBL_longest_isoforms.fa > ../../proteins_processed/SALRO.fasta
rm SALRO_ENSEMBL_longest_isoforms* SALRO.list
```

* STRMA

```
LongestIsoforms_ENSEMBL.py ../../proteins/STRMA_ENSEMBL_raw_pep.fa STRMA_ENSEMBL_longest_isoforms.fa
sed -i 's/^>/>STRMA_/' STRMA_ENSEMBL_longest_isoforms.fa
pymakeMap.py -gff STRMA_ENSEMBL.gff3 -p STRMA -f CDS -k protein_id -o ../chrom/STRMA.chrom -r fasta -F STRMA_ENSEMBL_longest_isoforms.fa
cut -f2 ../chrom/STRMA.chrom > STRMA.list ; pyfasta extract --file STRMA.list --header --space --fasta STRMA_ENSEMBL_longest_isoforms.fa > ../../proteins_processed/STRMA.fasta
rm STRMA_ENSEMBL_longest_isoforms* STRMA.list
```

* STRPU, we'll use `-d` flag of pymakemap to delete isoforms registered with different gene ids. It deletes 2891 proteins
This flag deletes overlapping genes with the following criteria:
* isoform with 1 exon is merged with overlapping isoform of 1+ size if any exon is overlapping on more than 20 base pairs
* isoform with 2 exons is merged with overlapping isoform of 2+ size if exons share at least a start or a stop
* isoform with 3 exons is merged with overlapping isoform of 3+ size if they share at least one internal exon


```
LongestIsoforms_ENSEMBL.py ../../proteins/STRPU_ENSEMBL_raw_pep.fa STRPU_ENSEMBL_longest_isoforms.fa
sed -i 's/^>/>STRPU_/' STRPU_ENSEMBL_longest_isoforms.fa
pymakeMap.py -gff STRPU_ENSEMBL.gff3 -p STRPU -f CDS -k protein_id -o ../chrom/STRPU.chrom -r fasta -F STRPU_ENSEMBL_longest_isoforms.fa
cut -f2 ../chrom/STRPU.chrom > STRPU.list ; pyfasta extract --file STRPU.list --header --space --fasta STRPU_ENSEMBL_longest_isoforms.fa > ../../proteins_processed/STRPU.fasta
rm STRPU_ENSEMBL_longest_isoforms* STRPU.list
```

* TRIAD

```
LongestIsoforms_ENSEMBL.py ../../proteins/TRIAD_ENSEMBL_raw_pep.fa TRIAD_ENSEMBL_longest_isoforms.fa
sed -i 's/^>/>TRIAD_/' TRIAD_ENSEMBL_longest_isoforms.fa
pymakeMap.py -gff TRIAD_ENSEMBL.gff3 -p TRIAD -f CDS -k protein_id -o ../chrom/TRIAD.chrom -r fasta -F TRIAD_ENSEMBL_longest_isoforms.fa
cut -f2 ../chrom/TRIAD.chrom > TRIAD.list ; pyfasta extract --file TRIAD.list --header --space --fasta TRIAD_ENSEMBL_longest_isoforms.fa > ../../proteins_processed/TRIAD.fasta
rm TRIAD_ENSEMBL_longest_isoforms* TRIAD.list
```

* TRICA

```
LongestIsoforms_ENSEMBL.py ../../proteins/TRICA_ENSEMBL_raw_pep.fa TRICA_ENSEMBL_longest_isoforms.fa
sed -i 's/^>/>TRICA_/' TRICA_ENSEMBL_longest_isoforms.fa
pymakeMap.py -gff TRICA_ENSEMBL.gff3 -p TRICA -f CDS -k protein_id -o ../chrom/TRICA.chrom -r fasta -F TRICA_ENSEMBL_longest_isoforms.fa
cut -f2 ../chrom/TRICA.chrom > TRICA.list ; pyfasta extract --file TRICA.list --header --space --fasta TRICA_ENSEMBL_longest_isoforms.fa > ../../proteins_processed/TRICA.fasta
rm TRICA_ENSEMBL_longest_isoforms* TRICA.list
```

#### 2.1.3 Prepare chrom files for the data from OTHER sources
##### 2.1.3.1 Prepare data with transcript variants
* HOIHO

```
LongestIsoforms_HOIHO_HYDVU_PTYFL.py ../../proteins/HOIHO_OTHER_raw_pep.fa HOIHO_OTHER_longest_isoforms.fa
sed -i 's/^>/>HOIHO_/' HOIHO_OTHER_longest_isoforms.fa
pymakeMap.py -gff HOIHO_OTHER.gff3 -p HOIHO -f CDS -k Name -o ../chrom/HOIHO.chrom -r fasta -F HOIHO_OTHER_longest_isoforms.fa
cut -f2 ../chrom/HOIHO.chrom > HOIHO.list ; pyfasta extract --file HOIHO.list --header --space --fasta HOIHO_OTHER_longest_isoforms.fa > ../../proteins_processed/HOIHO.fasta
rm HOIHO_OTHER_longest_isoforms* HOIHO.list
```

* HYDVU

```
LongestIsoforms_HOIHO_HYDVU_PTYFL.py ../../proteins/HYDVU_OTHER_raw_pep.fa HYDVU_OTHER_longest_isoforms.fa
sed -i 's/^>/>HYDVU_/' HYDVU_OTHER_longest_isoforms.fa
pymakeMap.py -gff HYDVU_OTHER.gff3 -p HYDVU -f CDS -k Parent -o ../chrom/HYDVU.chrom -r fasta -F HYDVU_OTHER_longest_isoforms.fa
cut -f2 ../chrom/HYDVU.chrom > HYDVU.list ; pyfasta extract --file HYDVU.list --header --space --fasta HYDVU_OTHER_longest_isoforms.fa > ../../proteins_processed/HYDVU.fasta
rm HYDVU_OTHER_longest_isoforms* HYDVU.list
```

* PTYFL

```
LongestIsoforms_HOIHO_HYDVU_PTYFL.py ../../proteins/PTYFL_OTHER_raw_pep.fa PTYFL_OTHER_longest_isoforms.fa
sed -i 's/^>/>PTYFL_/' PTYFL_OTHER_longest_isoforms.fa
pymakeMap.py -gff PTYFL_OTHER.gff3 -p PTYFL -f CDS -k Parent -o ../chrom/PTYFL.chrom -r fasta -F PTYFL_OTHER_longest_isoforms.fa
cut -f2 ../chrom/PTYFL.chrom > PTYFL.list ; pyfasta extract --file PTYFL.list --header --space --fasta PTYFL_OTHER_longest_isoforms.fa > ../../proteins_processed/PTYFL.fasta
rm PTYFL_OTHER_longest_isoforms* PTYFL.list
```

* SCHME

```
LongestIsoforms_SCHME.py ../../proteins/SCHME_OTHER_raw_pep.fa SCHME_OTHER.gff3 SCHME_OTHER_longest_isoforms.fa
sed -i 's/^>/>SCHME_/' SCHME_OTHER_longest_isoforms.fa
pymakeMap.py -gff SCHME_OTHER.gff3 -p SCHME -f CDS -k Parent -o ../chrom/SCHME.chrom -r fasta -F SCHME_OTHER_longest_isoforms.fa
cut -f2 ../chrom/SCHME.chrom > SCHME.list ; pyfasta extract --file SCHME.list --header --space --fasta SCHME_OTHER_longest_isoforms.fa > ../../proteins_processed/SCHME.fasta
rm SCHME_OTHER_longest_isoforms* SCHME.list
```

##### 2.1.3.2 Prepare data without transcript variants
For all the following animals, proteomes are already filtered for redundancy

* AURAU

```
sed 's/^>/>AURAU_/' ../../proteins/AURAU_OTHER_raw_pep.fa > AURAU_OTHER_longest_isoforms.fa
pymakeMap.py -gff AURAU_other.gff3 -p AURAU -f CDS -k protein_id -o ../chrom/AURAU.chrom -r fasta -F AURAU_OTHER_longest_isoforms.fa
cut -f2 ../chrom/AURAU.chrom > AURAU.list ; pyfasta extract --file AURAU.list --header --space --fasta AURAU_OTHER_longest_isoforms.fa > ../../proteins_processed/AURAU.fasta
rm AURAU_OTHER_longest_isoforms* AURAU.list
```

* CLYHE

```
sed 's/^>\([^-]*\)-protein/>CLYHE_\1/' ../../proteins/CLYHE_OTHER_raw_pep.fa > CLYHE_OTHER_longest_isoforms.fa
pymakeMap.py -gff CLYHE_OTHER.gff3 -p CLYHE -f exon -k Parent -o ../chrom/CLYHE.chrom -r fasta -F CLYHE_OTHER_longest_isoforms.fa
cut -f2 ../chrom/CLYHE.chrom > CLYHE.list ; pyfasta extract --file CLYHE.list --header --space --fasta CLYHE_OTHER_longest_isoforms.fa > ../../proteins_processed/CLYHE.fasta
rm CLYHE_OTHER_longest_isoforms* CLYHE.list
```

* EUPSC

```
sed 's/\tcluster_/\tID=cluster_/' esc_allchroms_uniq_15jul2019.gff > EUPSC.gff
pymakeMap.py -gff EUPSC.gff -p EUPSC -f exon -k ID -o ../chrom/EUPSC.chrom -r fasta -F EUPSC_OTHER_longest_isoforms.fa
cut -f2 ../chrom/EUPSC.chrom > EUPSC.list ; pyfasta extract --file EUPSC.list --header --space --fasta EUPSC_OTHER_longest_isoforms.fa > ../../proteins_processed/EUPSC.fasta
rm EUPSC_OTHER_longest_isoforms* EUPSC.list
```

* HOFMI

```
sed 's/^>\([^_]\+_[^_]\+\)_\(g[0-9]\+.t[0-9]\)/>HOFMI_\2\ \1/' ../../proteins/HOFMI_OTHER_raw_pep.fa > HOFMI_OTHER_longest_isoforms.fa
pymakeMap.py -gff HOFMI_OTHER.gff3 -p HOFMI -f CDS -k Parent -o ../chrom/HOFMI.chrom -r fasta -F HOFMI_OTHER_longest_isoforms.fa
cut -f2 ../chrom/HOFMI.chrom > HOFMI.list ; pyfasta extract --file HOFMI.list --header --space --fasta HOFMI_OTHER_longest_isoforms.fa > ../../proteins_processed/HOFMI.fasta
rm HOFMI_OTHER_longest_isoforms* HOFMI.list
```

* MIZYE

```
sed 's/^>/>MIZYE_/' ../../proteins/MIZYE_OTHER_raw_pep.fa > MIZYE_OTHER_longest_isoforms.fa
pymakeMap.py -gff SCALLOP/chr.id.gff3 -p MIZYE -f CDS -k Parent -o ../chrom/MIZYE.chrom -r fasta -F MIZYE_OTHER_longest_isoforms.fa
cut -f2 ../chrom/MIZYE.chrom > MIZYE.list ; pyfasta extract --file MIZYE.list --header --space --fasta MIZYE_OTHER_longest_isoforms.fa > ../../proteins_processed/MIZYE.fasta
rm MIZYE_OTHER_longest_isoforms* MIZYE.list
```

* MNELE

```
sed 's/^>/>MNELE_/' ../../proteins/MNELE_OTHER_raw_pep.fa > MNELE_OTHER_longest_isoforms.fa
pymakeMap.py -gff MNELE_OTHER.gff3 -p MNELE -f CDS -k Parent -o ../chrom/MNELE.chrom -r fasta -F MNELE_OTHER_longest_isoforms.fa
cut -f2 ../chrom/MNELE.chrom > MNELE.list ; pyfasta extract --file MNELE.list --header --space --fasta MNELE_OTHER_longest_isoforms.fa > ../../proteins_processed/MNELE.fasta
rm MNELE_OTHER_longest_isoforms* MNELE.list
```

* PLEBA

```
sed 's/^>/>PLEBA_/' ../../proteins/PLEBA_OTHER_raw_pep.fa > PLEBA_OTHER_longest_isoforms.fa
pymakeMap.py -gff PLEBA_mapped_final.gff3 -p PLEBA -f exon -k Name -o ../chrom/PLEBA.chrom -r fasta -F PLEBA_OTHER_longest_isoforms.fa
cut -f2 ../chrom/PLEBA.chrom > PLEBA.list ; pyfasta extract --file PLEBA.list --header --space --fasta PLEBA_OTHER_longest_isoforms.fa > ../../proteins_processed/PLEBA.fasta
rm PLEBA_OTHER_longest_isoforms* PLEBA.list
```

* SACKO

```
sed 's/^>/>SACKO_/' ../../proteins/SACKO_OTHER_raw_pep.fa > SACKO_OTHER_longest_isoforms.fa
pymakeMap.py -gff SACKO_prepped.gff3 -p SACKO -f CDS -k protein_id -o ../chrom/SACKO.chrom -r fasta -F SACKO_OTHER_longest_isoforms.fa
cut -f2 ../chrom/SACKO.chrom > SACKO.list ; pyfasta extract --file SACKO.list --header --space --fasta SACKO_OTHER_longest_isoforms.fa > ../../proteins_processed/SACKO.fasta
rm SACKO_OTHER_longest_isoforms* SACKO.list
```

* SYCCI

```
sed 's/^>/>SYCCI_/' ../../proteins/SYCCI_OTHER_raw_pep.fa > SYCCI_OTHER_longest_isoforms.fa
pymakeMap.py -gff SYCCI_mapped_final.gff3 -p SYCCI -f exon -k Name -o ../chrom/SYCCI.chrom -r fasta -F SYCCI_OTHER_longest_isoforms.fa -d
cut -f2 ../chrom/SYCCI.chrom > SYCCI.list ; pyfasta extract --file SYCCI.list --header --space --fasta SYCCI_OTHER_longest_isoforms.fa > ../../proteins_processed/SYCCI.fasta
rm SYCCI_OTHER_longest_isoforms* SYCCI.list
```
