## 5. Gene density analysis
### 5.1 Data preparation
#### 5.1.1 Download genome files
If possible, sequences have been downloaded as softmasked (repetitive sequences in lowercase). This is more informative than hardmask (all residues in uppercase, masked residues replaces by Ns).

```
#From NCBI: 16 genomes
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/405/GCF_000001405.39_GRCh38.p13/GCF_000001405.39_GRCh38.p13_genomic.fna.gz -o HOMSA.masked_genome.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/635/GCF_000001635.26_GRCm38.p6/GCF_000001635.26_GRCm38.p6_genomic.fna.gz -o MUSMU.masked_genome.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/035/GCF_000002035.6_GRCz11/GCF_000002035.6_GRCz11_genomic.fna.gz -o DANRE.masked_genome.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/315/GCF_000002315.6_GRCg6a/GCF_000002315.6_GRCg6a_genomic.fna.gz -o GALGA.masked_genome.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/004/195/GCF_000004195.3_Xenopus_tropicalis_v9.1/GCF_000004195.3_Xenopus_tropicalis_v9.1_genomic.fna.gz -o XENTR.masked_genome.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/165/045/GCF_000165045.1_Callorhinchus_milii-6.1.3/GCF_000165045.1_Callorhinchus_milii-6.1.3_genomic.fna.gz -o CALMI.masked_genome.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/004/143/615/GCF_004143615.1_amil_sf_1.1/GCF_004143615.1_amil_sf_1.1_genomic.fna.gz -o ACRMI.masked_genome.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/224/145/GCF_000224145.3_KH/GCF_000224145.3_KH_genomic.fna.gz -o CIOIN.masked_genome.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/225/785/GCF_000225785.1_LatCha1/GCF_000225785.1_LatCha1_genomic.fna.gz -o LATCH.masked_genome.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/238/955/GCF_000238955.4_M_zebra_UMD2a/GCF_000238955.4_M_zebra_UMD2a_genomic.fna.gz -o MAYZE.masked_genome.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/242/695/GCF_000242695.1_LepOcu1/GCF_000242695.1_LepOcu1_genomic.fna.gz -o LEPOC.masked_genome.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/344/595/GCF_000344595.1_CheMyd_1.0/GCF_000344595.1_CheMyd_1.0_genomic.fna.gz -o CHEMY.masked_genome.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/365/465/GCF_000365465.2_Ptep_2.0/GCF_000365465.2_Ptep_2.0_genomic.fna.gz -o PARTE.masked_genome.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/001/417/965/GCF_001417965.1_Aiptasia_genome_1.1/GCF_001417965.1_Aiptasia_genome_1.1_genomic.fna.gz -o EXAPA.masked_genome.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/001/891/065/GCF_001891065.1_H_comes_QL1_v1/GCF_001891065.1_H_comes_QL1_v1_genomic.fna.gz -o HIPCO.masked_genome.gz
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/001/949/145/GCF_001949145.1_OKI-Apl_1.0/GCF_001949145.1_OKI-Apl_1.0_genomic.fna.gz -o ACAPL.masked_genome.gz

#From ENSEMBL: 20 genomes
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/fasta/adineta_vaga/dna/Adineta_vaga.AMS_PRJEB1171_v1.dna_sm.toplevel.fa.gz -o ADIVA.masked_genome.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/fasta/amphimedon_queenslandica/dna/Amphimedon_queenslandica.Aqu1.dna_sm.toplevel.fa.gz -o AMPQU.masked_genome.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/fasta/anopheles_gambiae/dna/Anopheles_gambiae.AgamP4.dna_sm.toplevel.fa.gz -o ANOGA.masked_genome.gz
curl ftp://ftp.ensemblgenomes.org/pub/release-45/metazoa/fasta/branchiostoma_lanceolatum/dna/Branchiostoma_lanceolatum.BraLan2.dna_sm.toplevel.fa.gz -o BRALA.masked_genome.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/fasta/caenorhabditis_elegans/dna/Caenorhabditis_elegans.WBcel235.dna_sm.toplevel.fa.gz -o CAEEL.masked_genome.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/fasta/capitella_teleta/dna/Capitella_teleta.Capitella_teleta_v1.0.dna_sm.toplevel.fa.gz -o CAPTE.masked_genome.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/fasta/crassostrea_gigas/dna/Crassostrea_gigas.oyster_v9.dna_sm.toplevel.fa.gz -o CRAGI.masked_genome.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/fasta/daphnia_pulex/dna/Daphnia_pulex.V1.0.dna_sm.toplevel.fa.gz -o DAPPU.masked_genome.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/fasta/drosophila_melanogaster/dna/Drosophila_melanogaster.BDGP6.22.dna_sm.toplevel.fa.gz -o DROME.masked_genome.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/fasta/helobdella_robusta/dna/Helobdella_robusta.Helro1.dna_sm.toplevel.fa.gz -o HELRO.masked_genome.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/fasta/ixodes_scapularis/dna/Ixodes_scapularis.IscaW1.dna_sm.toplevel.fa.gz -o IXOSC.masked_genome.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/fasta/lingula_anatina/dna/Lingula_anatina.LinAna1.0.dna_sm.toplevel.fa.gz -o LINAN.masked_genome.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/fasta/lottia_gigantea/dna/Lottia_gigantea.Lotgi1.dna_sm.toplevel.fa.gz -o LOTGI.masked_genome.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/fasta/nematostella_vectensis/dna/Nematostella_vectensis.ASM20922v1.dna_sm.toplevel.fa.gz -o NEMVE.masked_genome.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/fasta/strigamia_maritima/dna/Strigamia_maritima.Smar1.dna_sm.toplevel.fa.gz -o STRMA.masked_genome.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/fasta/strongylocentrotus_purpuratus/dna/Strongylocentrotus_purpuratus.Spur_3.1.dna_sm.toplevel.fa.gz -o STRPU.masked_genome.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/fasta/tribolium_castaneum/dna/Tribolium_castaneum.Tcas5.2.dna_sm.toplevel.fa.gz -o TRICA.masked_genome.gz
curl ftp://ftp.ensemblgenomes.org/pub/metazoa/release-45/fasta/trichoplax_adhaerens/dna/Trichoplax_adhaerens.ASM15027v1.dna_sm.toplevel.fa.gz -o TRIAD.masked_genome.gz
curl ftp://ftp.ensemblgenomes.org/pub/protists/release-45/fasta/protists_choanoflagellida1_collection/salpingoeca_rosetta_gca_000188695/dna/Salpingoeca_rosetta_gca_000188695.Proterospongia_sp_ATCC50818.dna_sm.toplevel.fa.gz -o SALRO.masked_genome.gz
curl ftp://ftp.ensemblgenomes.org/pub/protists/release-45/fasta/protists_ichthyosporea1_collection/capsaspora_owczarzaki_atcc_30864_gca_000151315/dna/Capsaspora_owczarzaki_atcc_30864_gca_000151315.C_owczarzaki_V2.dna_sm.toplevel.fa.gz -o CAPOW.masked_genome.gz

#From other sources: 7 genomes
curl http://marimba.obs-vlfr.fr/download/file/fid/48 -o CLYHE.hardmasked_genome.gz
curl https://research.nhgri.nih.gov/hydra/download/assembly/Hm105_Dovetail_Assembly_1.0.fa.gz -o HYDVU.masked_genome.gz
curl https://bitbucket.org/molpalmuc/hoilungia-genome/raw/0d523a5b8556741a37918f3f30d0ed0414833912/sequences/Hhon_final_contigs_softmasked.fasta.gz -o HOIHO.masked_genome.gz
curl https://research.nhgri.nih.gov/mnemiopsis/download/genome/MlScaffold09.nt.gz -o MNELE.masked_genome.gz
curl https://marinegenomics.oist.jp/acornworm/download/pfl_scaffold_ver1.0.14.masked.fasta.gz -o PTYFL.masked_genome.gz
curl http://planmine.mpi-cbg.de/planmine/model/bulkdata/dd_Smes_g4.fasta.zip -o SCHME.masked_genome
curl ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/695/325/GCA_000695325.1_P.bachei_draft_genome_v.1.1/GCA_000695325.1_P.bachei_draft_genome_v.1.1_genomic.fna.gz -o PLEBA.masked_genome
```

The 3 remaining genomes were downloaded with a browser (and or made using annotation files) or used from another emplacment in cube:

* Aurelia aurita from: https://drive.google.com/drive/folders/1NC6bZ9cxWkZyofOsMPzrxIH3C7m1ySiu, Aurelia.Genome_v1.2_Protein_Models_12-28-18.fasta, renamed AURAU.hardmasked_genome
* Saccoglossus kowalevskii: SkowalevskiiJGIv3.0.longestTrs.pep.fa.gz Downloaded from Metazome v3 (https://metazome.jgi.doe.gov/pz/portal.html#!bulk?org=Org_Skowalevskii_er), called SACKO.masked_genome.gz
* Sycon ciliatum genome, CDS and peptide downloaded from datadryad (https://datadryad.org/resource/doi:10.5061/dryad.tn0f3).


EUPSC.masked_genome genome from Schmidbaur et al. in prep.

Finally, we gunzip all the gz files 

```
gunzip *.fa.gz
```
## 5 Gene density analysis (REDUX)
### 5.1. reconstructing ancestral counts of syntenic blocks
We'll use 5 alternating topologies:

* Ctenophore-sister (CS) (cteno,(sponge,(plac,(cnid,(acoel,prot,deut)))));
* Sponge-sister (PS) (sponge,(cteno,(plac,(cnid,(acoel,prot,deut))))); 
* Xenacoelomorpha (XNEP): (cteno,sponge,(plac,(cnid,(acoel(prot,deut))))));  
* Xenambulacraria (XAMB): (cteno,sponge,(plac,(cnid,(prot((chord,(acoel,ambulacraria)))))));
* Consensus (CONS) (cteno,sponge,(plac,(cnid,(acoel,prot,deut))));

Here are the actual trees:

* CONS
```
(((((((((((((((HOMSA,MUSMU),(CHEMY,GALGA)),XENTR),LATCH),(((MAYZE,HIPCO),DANRE),LEPOC)),CALMI)Vertebrata,CIOIN)Olfactores,BRALA)Chordata,((SACKO,PTYFL),(STRPU,ACAPL))Ambulacraria)Deuterostomia,(((EUPSC,LOTGI,(MIZYE,CRAGI))Mollusca,(CAPTE,HELRO),ADIVA,LINAN,SCHME)Lophotrochozoa,((((((DROME,ANOGA),TRICA),DAPPU),STRMA),(IXOSC,PARTE)),CAEEL)Ecdysozoa)Protostomia,HOFMI)Bilateria,(((NEMVE,EXAPA),ACRMI),((HYDVU,CLYHE),AURAU))Cnidaria)Planulozoa,(HOIHO,TRIAD)Placozoa)Parahoxozoa,(PLEBA,MNELE)Ctenophora,(SYCCI,AMPQU)Porifera)Metazoa,SALRO)Choanozoa,CAPOW)Filozoa;
```
* CS
```
((((((((((((((((HOMSA,MUSMU),(CHEMY,GALGA)),XENTR),LATCH),(((MAYZE,HIPCO),DANRE),LEPOC)),CALMI)Vertebrata,CIOIN)Olfactores,BRALA)Chordata,((SACKO,PTYFL),(STRPU,ACAPL))Ambulacraria)Deuterostomia,(((EUPSC,LOTGI,(MIZYE,CRAGI))Mollusca,(CAPTE,HELRO),ADIVA,LINAN,SCHME)Lophotrochozoa,((((((DROME,ANOGA),TRICA),DAPPU),STRMA),(IXOSC,PARTE)),CAEEL)Ecdysozoa)Protostomia,HOFMI)Bilateria,(((NEMVE,EXAPA),ACRMI),((HYDVU,CLYHE),AURAU))Cnidaria)Planulozoa,(HOIHO,TRIAD)Placozoa)Parahoxozoa,(SYCCI,AMPQU)Porifera)Porifera_Parahoxozoa,(PLEBA,MNELE)Ctenophora)Metazoa,SALRO)Choanozoa,CAPOW)Filozoa;
```
* PS
```
((((((((((((((((HOMSA,MUSMU),(CHEMY,GALGA)),XENTR),LATCH),(((MAYZE,HIPCO),DANRE),LEPOC)),CALMI)Vertebrata,CIOIN)Olfactores,BRALA)Chordata,((SACKO,PTYFL),(STRPU,ACAPL))Ambulacraria)Deuterostomia,(((EUPSC,LOTGI,(MIZYE,CRAGI))Mollusca,(CAPTE,HELRO),ADIVA,LINAN,SCHME)Lophotrochozoa,((((((DROME,ANOGA),TRICA),DAPPU),STRMA),(IXOSC,PARTE)),CAEEL)Ecdysozoa)Protostomia,HOFMI)Bilateria,(((NEMVE,EXAPA),ACRMI),((HYDVU,CLYHE),AURAU))Cnidaria)Planulozoa,(HOIHO,TRIAD)Placozoa)Parahoxozoa,(PLEBA,MNELE)Ctenophora)Ctenophora_Parahoxozoa,(SYCCI,AMPQU)Porifera)Metazoa,SALRO)Choanozoa,CAPOW)Filozoa;
```
* XAMB
```
(((((((((((((((HOMSA,MUSMU),(CHEMY,GALGA)),XENTR),LATCH),(((MAYZE,HIPCO),DANRE),LEPOC)),CALMI)Vertebrata,CIOIN)Olfactores,BRALA)Chordata,(HOFMI,((SACKO,PTYFL),(STRPU,ACAPL))Ambulacraria)Xenambulacraria)Deuterostomia,(((EUPSC,LOTGI,(MIZYE,CRAGI))Mollusca,(CAPTE,HELRO),ADIVA,LINAN,SCHME)Lophotrochozoa,((((((DROME,ANOGA),TRICA),DAPPU),STRMA),(IXOSC,PARTE)),CAEEL)Ecdysozoa)Protostomia)Bilateria,(((NEMVE,EXAPA),ACRMI),((HYDVU,CLYHE),AURAU))Cnidaria)Planulozoa,(HOIHO,TRIAD)Placozoa)Parahoxozoa,(PLEBA,MNELE)Ctenophora,(SYCCI,AMPQU)Porifera)Metazoa,SALRO)Choanozoa,CAPOW)Filozoa;
```

* XNEP
```
((((((((((((((((HOMSA,MUSMU),(CHEMY,GALGA)),XENTR),LATCH),(((MAYZE,HIPCO),DANRE),LEPOC)),CALMI)Vertebrata,CIOIN)Olfactores,BRALA)Chordata,((SACKO,PTYFL),(STRPU,ACAPL))Ambulacraria)Deuterostomia,(((EUPSC,LOTGI,(MIZYE,CRAGI))Mollusca,(CAPTE,HELRO),ADIVA,LINAN,SCHME)Lophotrochozoa,((((((DROME,ANOGA),TRICA),DAPPU),STRMA),(IXOSC,PARTE)),CAEEL)Ecdysozoa)Protostomia)Nephrozoa,HOFMI)Bilateria,(((NEMVE,EXAPA),ACRMI),((HYDVU,CLYHE),AURAU))Cnidaria)Planulozoa,(HOIHO,TRIAD)Placozoa)Parahoxozoa,(PLEBA,MNELE)Ctenophora,(SYCCI,AMPQU)Porifera)Metazoa,SALRO)Choanozoa,CAPOW)Filozoa;
```

CONS is the tree we want to use for everything (since  alternate topologies generate similar results).
CS and PS will be used only for checking that the polytomies in our tree have no incidence on the estimation of metazoan novelties.
XNEP and XAMB are to be used for checking that the polytomies in our tree don't affect estimation of novelties in  Bilateria/Nephrozoa (LCA of protostomes and deuterostomes...) and deuterostomia LCA.
And ofc, the initial multifurcating tree, polytomic at the base of metazoa and at the base of bilateria

`filt.clusters` is filtered out version of the total multi_species block (minus cluster 1, 8406 blocks considered as one multi species block, due to fusing orthologous groups together. Randomized chromosomes don't have such artifacts, no need to filter them.

* Ctenophore-sister
```
cd 02_gene_density_analysis
python3 BlocksByNode.py -c filt.clusters -b ../01_microsynteny/chrom_of/5.blocks.3.syn_corrected.synt -s trees/cteno_sister_bila_poly.tre -n Choanozoa Metazoa Porifera_Parahoxozoa Parahoxozoa Planulozoa Cnidaria Bilateria Deuterostomia Protostomia Chordata Ambulacraria Lophotrochozoa Ecdysozoa Olfactores Mollusca Vertebrata -m 2 -r short -t ancestral novel

#clusters list, to draw Venn diagrams of clusters found in nodes
python3 BlocksByNode.py -c filt.clusters -b ../01_microsynteny/chrom_of/5.blocks.3.syn_corrected.synt -s trees/cteno_sister_bila_poly.tre -n Metazoa Porifera_Parahoxozoa -m 2 -r clusters_list -t novel|cut -f1,2 > venn_diag/CS.tsv



#counts for randomized blocks
python3 BlocksByNode.py -c ../01_microsynteny/randomized_chrom/rand.1/Orthofinder/5.blocks.3.syn.clusters -b ../01_microsynteny/randomized_chrom/rand.1/Orthofinder/5.blocks.3.syn_corrected.synt -s trees/cteno_sister_bila_poly.tre -n Choanozoa Metazoa Porifera_Parahoxozoa Parahoxozoa Planulozoa Cnidaria Bilateria Deuterostomia Protostomia Chordata Ambulacraria Lophotrochozoa Ecdysozoa Olfactores Mollusca Vertebrata -m 2 -r short -t ancestral novel
python3 BlocksByNode.py -c ../01_microsynteny/randomized_chrom/rand.2/Orthofinder/5.blocks.3.syn.clusters -b ../01_microsynteny/randomized_chrom/rand.2/Orthofinder/5.blocks.3.syn_corrected.synt -s trees/cteno_sister_bila_poly.tre -n Choanozoa Metazoa Porifera_Parahoxozoa Parahoxozoa Planulozoa Cnidaria Bilateria Deuterostomia Protostomia Chordata Ambulacraria Lophotrochozoa Ecdysozoa Olfactores Mollusca Vertebrata -m 2 -r short -t ancestral novel
python3 BlocksByNode.py -c ../01_microsynteny/randomized_chrom/rand.3/Orthofinder/5.blocks.3.syn.clusters -b ../01_microsynteny/randomized_chrom/rand.3/Orthofinder/5.blocks.3.syn_corrected.synt -s trees/cteno_sister_bila_poly.tre -n Choanozoa Metazoa Porifera_Parahoxozoa Parahoxozoa Planulozoa Cnidaria Bilateria Deuterostomia Protostomia Chordata Ambulacraria Lophotrochozoa Ecdysozoa Olfactores Mollusca Vertebrata -m 2 -r short -t ancestral novel

```

Ctenophore-sister    | ancestral | novel | recency
---------------------|-----------|-------|--------
Choanozoa            |    17     |   14  |    1
Metazoa              |     0     |    9  |    2
Porifera_Parahoxozoa |     9     |   25  |    3
Parahoxozoa          |    50     |    6  |    4
Planulozoa           |    63     |  162  |    5
Cnidaria             |   254     |   11  |    6
Bilateria            |   335     |  256  |    6
Deuterostomia        |   545     |    3  |    7
Protostomia          |   612     |   16  |    7
Chordata             |   511     |    1  |    8
Ambulacraria         |   430     |    6  |    8
Lophotrochozoa       |   798     |   91  |    8
Ecdysozoa            |   241     |    1  |    8
Olfactores           |   147     |    1  |    9
Mollusca             |   662     |   66  |    9
Vertebrata           |    70     |  170  |   10



* Porifera-sister
```
python3 BlocksByNode.py -c filt.clusters -b ../01_microsynteny/chrom_of/5.blocks.3.syn_corrected.synt -s trees/sponge_sister_bila_poly.tre -n Choanozoa Metazoa Ctenophora_Parahoxozoa Parahoxozoa Planulozoa Cnidaria Bilateria Deuterostomia Protostomia Chordata Ambulacraria Lophotrochozoa Ecdysozoa Olfactores Mollusca Vertebrata -m 2 -r short -t ancestral novel

#clusters list, to draw Venn diagrams of clusters found in nodes
python3 BlocksByNode.py -c filt.clusters -b ../01_microsynteny/chrom_of/5.blocks.3.syn_corrected.synt -s trees/sponge_sister_bila_poly.tre -n Metazoa Ctenophora_Parahoxozoa -m 2 -r clusters_list -t novel|cut -f1,2 > venn_diag/PS.tsv

#counts for randomized blocks
python3 BlocksByNode.py -c ../01_microsynteny/randomized_chrom/rand.1/Orthofinder/5.blocks.3.syn.clusters -b ../01_microsynteny/randomized_chrom/rand.1/Orthofinder/5.blocks.3.syn_corrected.synt -s trees/sponge_sister_bila_poly.tre -n Choanozoa Metazoa Ctenophora_Parahoxozoa Parahoxozoa Planulozoa Cnidaria Bilateria Deuterostomia Protostomia Chordata Ambulacraria Lophotrochozoa Ecdysozoa Olfactores Mollusca Vertebrata -m 2 -r short -t ancestral novel
python3 BlocksByNode.py -c ../01_microsynteny/randomized_chrom/rand.2/Orthofinder/5.blocks.3.syn.clusters -b ../01_microsynteny/randomized_chrom/rand.2/Orthofinder/5.blocks.3.syn_corrected.synt -s trees/sponge_sister_bila_poly.tre -n Choanozoa Metazoa Ctenophora_Parahoxozoa Parahoxozoa Planulozoa Cnidaria Bilateria Deuterostomia Protostomia Chordata Ambulacraria Lophotrochozoa Ecdysozoa Olfactores Mollusca Vertebrata -m 2 -r short -t ancestral novel
python3 BlocksByNode.py -c ../01_microsynteny/randomized_chrom/rand.3/Orthofinder/5.blocks.3.syn.clusters -b ../01_microsynteny/randomized_chrom/rand.3/Orthofinder/5.blocks.3.syn_corrected.synt -s trees/sponge_sister_bila_poly.tre -n Choanozoa Metazoa Ctenophora_Parahoxozoa Parahoxozoa Planulozoa Cnidaria Bilateria Deuterostomia Protostomia Chordata Ambulacraria Lophotrochozoa Ecdysozoa Olfactores Mollusca Vertebrata -m 2 -r short -t ancestral novel
```

Porifera-sister        | ancestral | novel | recency
-----------------------|-----------|-------|--------
Choanozoa              |    17     |   14  |    1
Metazoa                |     0     |   25  |    2
Ctenophora_Parahoxozoa |    31     |    7  |    3
Parahoxozoa            |    50     |    6  |    4
Planulozoa             |    63     |  162  |    5
Cnidaria               |   254     |   11  |    6
Bilateria              |   335     |  256  |    6
Deuterostomia          |   545     |    3  |    7
Protostomia            |   612     |   16  |    7
Chordata               |   511     |    1  |    8
Ambulacraria           |   430     |    6  |    8
Lophotrochozoa         |   798     |   91  |    8
Ecdysozoa              |   241     |    1  |    8
Olfactores             |   147     |    1  |    9
Mollusca               |   662     |   66  |    9
Vertebrata             |    70     |  170  |   10



* Xenacoelomorpha

```
python3 BlocksByNode.py -c filt.clusters -b ../01_microsynteny/chrom_of/5.blocks.3.syn_corrected.synt -s trees/meta_poly_xenacoelomorpha.tre -n Choanozoa Metazoa Parahoxozoa Planulozoa Cnidaria Bilateria Nephrozoa Deuterostomia Protostomia Chordata Ambulacraria Lophotrochozoa Ecdysozoa Olfactores Mollusca Vertebrata -m 2 -r short -t ancestral novel

#clusters list, to draw Venn diagrams of clusters found in nodes
python3 BlocksByNode.py -c filt.clusters -b ../01_microsynteny/chrom_of/5.blocks.3.syn_corrected.synt -s trees/meta_poly_xenacoelomorpha.tre -n Bilateria Nephrozoa Deuterostomia Ambulacraria -m 2 -r clusters_list -t novel|cut -f1,2 > venn_diag/XNEP.tsv

#count blocks in randomized genomes
python3 BlocksByNode.py -c ../01_microsynteny/randomized_chrom/rand.1/Orthofinder/5.blocks.3.syn.clusters -b ../01_microsynteny/randomized_chrom/rand.1/Orthofinder/5.blocks.3.syn_corrected.synt -s trees/meta_poly_xenacoelomorpha.tre -n Choanozoa Metazoa Parahoxozoa Planulozoa Cnidaria Bilateria Nephrozoa Deuterostomia Protostomia Chordata Ambulacraria Lophotrochozoa Ecdysozoa Olfactores Mollusca Vertebrata -m 2 -r short -t ancestral novel
python3 BlocksByNode.py -c ../01_microsynteny/randomized_chrom/rand.2/Orthofinder/5.blocks.3.syn.clusters -b ../01_microsynteny/randomized_chrom/rand.2/Orthofinder/5.blocks.3.syn_corrected.synt -s trees/meta_poly_xenacoelomorpha.tre -n Choanozoa Metazoa Parahoxozoa Planulozoa Cnidaria Bilateria Nephrozoa Deuterostomia Protostomia Chordata Ambulacraria Lophotrochozoa Ecdysozoa Olfactores Mollusca Vertebrata -m 2 -r short -t ancestral novel
python3 BlocksByNode.py -c ../01_microsynteny/randomized_chrom/rand.3/Orthofinder/5.blocks.3.syn.clusters -b ../01_microsynteny/randomized_chrom/rand.3/Orthofinder/5.blocks.3.syn_corrected.synt -s trees/meta_poly_xenacoelomorpha.tre -n Choanozoa Metazoa Parahoxozoa Planulozoa Cnidaria Bilateria Nephrozoa Deuterostomia Protostomia Chordata Ambulacraria Lophotrochozoa Ecdysozoa Olfactores Mollusca Vertebrata -m 2 -r short -t ancestral novel

```

Xenacoelomorpha        | ancestral | novel | recency
-----------------------|-----------|-------|--------
Choanozoa              |    17     |   14  |    1
Metazoa                |     0     |   34  |    2
Parahoxozoa            |    50     |    6  |    3
Planulozoa             |    60     |  162  |    4
Cnidaria               |   254     |   11  |    5
Bilateria              |   335     |   38  |    5
Nephrozoa              |   354     |  224  |    6
Deuterostomia          |   545     |    3  |    7
Protostomia            |   612     |   16  |    7
Chordata               |   511     |    1  |    8
Ambulacraria           |   430     |    6  |    8
Lophotrochozoa         |   798     |   91  |    8
Ecdysozoa              |   241     |    1  |    8
Olfactores             |   147     |    1  |    9
Mollusca               |   662     |   66  |    9
Vertebrata             |    70     |  170  |   10

* Xenambulacraria

```
python3 BlocksByNode.py -c filt.clusters -b ../01_microsynteny/chrom_of/5.blocks.3.syn_corrected.synt -s trees/meta_poly_xenambulacraria.tre -n Choanozoa Metazoa Parahoxozoa Planulozoa Cnidaria Bilateria Deuterostomia Protostomia Chordata Xenambulacraria Ambulacraria Lophotrochozoa Ecdysozoa Olfactores Mollusca Vertebrata -m 2 -r short -t ancestral novel

#clusters list, to draw Venn diagrams of clusters found in nodes
python3 BlocksByNode.py -c filt.clusters -b ../01_microsynteny/chrom_of/5.blocks.3.syn_corrected.synt -s trees/meta_poly_xenambulacraria.tre -n Bilateria Deuterostomia Xenambulacraria Ambulacraria -m 2 -r clusters_list -t novel|cut -f1,2 > venn_diag/XAMB.tsv

#Isolate blocks, and clusters



#random block counts
python3 BlocksByNode.py -c ../01_microsynteny/randomized_chrom/rand.1/Orthofinder/5.blocks.3.syn.clusters -b ../01_microsynteny/randomized_chrom/rand.1/Orthofinder/5.blocks.3.syn_corrected.synt -s trees/meta_poly_xenambulacraria.tre -n Choanozoa Metazoa Parahoxozoa Planulozoa Cnidaria Bilateria Deuterostomia Protostomia Chordata Xenambulacraria Ambulacraria Lophotrochozoa Ecdysozoa Olfactores Mollusca Vertebrata -m 2 -r short -t ancestral novel
python3 BlocksByNode.py -c ../01_microsynteny/randomized_chrom/rand.2/Orthofinder/5.blocks.3.syn.clusters -b ../01_microsynteny/randomized_chrom/rand.2/Orthofinder/5.blocks.3.syn_corrected.synt -s trees/meta_poly_xenambulacraria.tre -n Choanozoa Metazoa Parahoxozoa Planulozoa Cnidaria Bilateria Deuterostomia Protostomia Chordata Xenambulacraria Ambulacraria Lophotrochozoa Ecdysozoa Olfactores Mollusca Vertebrata -m 2 -r short -t ancestral novel
python3 BlocksByNode.py -c ../01_microsynteny/randomized_chrom/rand.3/Orthofinder/5.blocks.3.syn.clusters -b ../01_microsynteny/randomized_chrom/rand.3/Orthofinder/5.blocks.3.syn_corrected.synt -s trees/meta_poly_xenambulacraria.tre -n Choanozoa Metazoa Parahoxozoa Planulozoa Cnidaria Bilateria Deuterostomia Protostomia Chordata Xenambulacraria Ambulacraria Lophotrochozoa Ecdysozoa Olfactores Mollusca Vertebrata -m 2 -r short -t ancestral novel


```
No syntenic novelty retained in xenacoelomorpha

Xenambulacraria        | ancestral | novel | recency
-----------------------|-----------|-------|--------
Choanozoa              |    17     |   14  |    1
Metazoa                |     0     |   34  |    2
Parahoxozoa            |    50     |    6  |    3
Planulozoa             |   63     |  162  |    4
Cnidaria               |   254     |   11  |    5
Bilateria              |   328     |  237  |    5
Deuterostomia          |   568     |    3  |    6
Protostomia            |   612     |   16  |    6
Xenambulacraria        |   511     |    0  |    7
Chordata               |   517     |    1  |    8
Ambulacraria           |   430     |    6  |    8
Lophotrochozoa         |   798     |   91  |    8
Ecdysozoa              |   241     |    1  |    8
Olfactores             |   147     |    1  |    9
Mollusca               |   662     |   66  |    9
Vertebrata             |    70     |  170  |   10


* Consensus
```
python3 BlocksByNode.py -c filt.clusters -b ../01_microsynteny/chrom_of/5.blocks.3.syn_corrected.synt -s trees/meta_poly_bila_poly.tre -n Choanozoa Metazoa Parahoxozoa Planulozoa Cnidaria Bilateria Deuterostomia Protostomia Chordata Ambulacraria Lophotrochozoa Ecdysozoa Olfactores Mollusca Vertebrata -m 2 -r short -t ancestral novel

#clusters list, to draw Venn diagrams of clusters found in nodes
python3 BlocksByNode.py -c filt.clusters -b ../01_microsynteny/chrom_of/5.blocks.3.syn_corrected.synt -s trees/meta_poly_bila_poly.tre -n Metazoa Bilateria Deuterostomia Ambulacraria -m 2 -r clusters_list -t novel|cut -f1,2 > venn_diag/CONS.tsv


nodelist=(Bilateria Lophotrochozoa Metazoa Parahoxozoa Planulozoa Vertebrata)
#Block types
for node in "${nodelist[@]}"; do
    python3 BlocksByNode.py -c filt.clusters -b ../01_microsynteny/chrom_of/5.blocks.3.syn_corrected.synt -s trees/meta_poly_bila_poly.tre -n $node -m 2 -r blocks_list -t novel|cut -f2- > bynode/CONS/$node.novel.synt
    python3 BlocksByNode.py -c filt.clusters -b ../01_microsynteny/chrom_of/5.blocks.3.syn_corrected.synt -s trees/meta_poly_bila_poly.tre -n $node -m 2 -r clusters_list -t novel|cut -f1,3- > bynode/CONS/$node.novel.clusters;
    done

#random block counts
python3 BlocksByNode.py -c ../01_microsynteny/randomized_chrom/rand.1/Orthofinder/5.blocks.3.syn.clusters -b ../01_microsynteny/randomized_chrom/rand.1/Orthofinder/5.blocks.3.syn_corrected.synt -s trees/meta_poly_bila_poly.tre -n Choanozoa Metazoa Parahoxozoa Planulozoa Cnidaria Bilateria Deuterostomia Protostomia Chordata Ambulacraria Lophotrochozoa Ecdysozoa Olfactores Mollusca Vertebrata -m 2 -r short -t ancestral novel
python3 BlocksByNode.py -c ../01_microsynteny/randomized_chrom/rand.2/Orthofinder/5.blocks.3.syn.clusters -b ../01_microsynteny/randomized_chrom/rand.2/Orthofinder/5.blocks.3.syn_corrected.synt -s trees/meta_poly_bila_poly.tre -n Choanozoa Metazoa Parahoxozoa Planulozoa Cnidaria Bilateria Deuterostomia Protostomia Chordata Ambulacraria Lophotrochozoa Ecdysozoa Olfactores Mollusca Vertebrata -m 2 -r short -t ancestral novel
python3 BlocksByNode.py -c ../01_microsynteny/randomized_chrom/rand.3/Orthofinder/5.blocks.3.syn.clusters -b ../01_microsynteny/randomized_chrom/rand.3/Orthofinder/5.blocks.3.syn_corrected.synt -s trees/meta_poly_bila_poly.tre -n Choanozoa Metazoa Parahoxozoa Planulozoa Cnidaria Bilateria Deuterostomia Protostomia Chordata Ambulacraria Lophotrochozoa Ecdysozoa Olfactores Mollusca Vertebrata -m 2 -r short -t ancestral novel
```

Consensus              | ancestral | novel | recency
-----------------------|-----------|-------|--------
Choanozoa              |    17     |   14  |    1
Metazoa                |     0     |   34  |    2
Parahoxozoa            |    50     |    6  |    3
Planulozoa             |    63     |  162  |    4
Cnidaria               |   254     |   11  |    5
Bilateria              |   335     |  256  |    5
Deuterostomia          |   545     |    3  |    6
Protostomia            |   612     |   16  |    6
Chordata               |   511     |    1  |    7
Ambulacraria           |   430     |    6  |    7
Lophotrochozoa         |   798     |   91  |    7
Ecdysozoa              |   241     |    1  |    7
Olfactores             |   147     |    1  |    8
Mollusca               |   662     |   66  |    8
Vertebrata             |    70     |  170  |    9




### 5.2 Measuring gene density at key nodes
We'll use consensus tree

```
cd 02_gene_density_analysis/bynode/CONS
#randomized blocks
nodelist=(Bilateria Lophotrochozoa Metazoa Parahoxozoa Planulozoa Vertebrata)
for node in "${nodelist[@]}"; do
    while read species; do
    echo "Rscript pick_random_blocks.R --only-genome=$species -n 100 random/$node/ $node.novel.synt ../../../01_microsynteny/chrom/$species.chrom";
    done < $node.species;
done > random.sh

cat random.sh |slurmtasks -n random_blocks |sbatch

for node in "${nodelist[@]}"; do
    cat random/$node/*tsv|grep -v "block_id" > $node.novel.random.synt ;
done
```

Now for measuring gene density in every animal and node, we use `make_tidy_density_df`.  It can take a while (2-3 hours for the 49 species dataset, 6 different syntenic nodes, everything runs sequentially on a single core of an AMD Opteron 6320).
This outputs a huge table with gene density by block, genome size, median intergenic distance, node, taxon, species, gene list, para/not_para classification (>40% or less than  40% of genes belonging to the same OG) (i.e. all info needed to build figures).

Total assembly size is measured by parsing fasta files
```
python3 make_tidy_density_df.py -s ../bynode/CONS/ -g ../genomes/ -c ../../01_microsynteny/chrom/ -m ../../01_microsynteny/chrom_of/5.blocks.3.syn.clusters -og ../../01_microsynteny/Orthofinder.clus
```
Make data to be used for scatterplots of SF4 (run in the same folder as tidy density df)
```
figure2_prep_data_scatterplots.py
```