## 6 GO_terms enrichment analysis
We'll use eggnogmapper.
Two steps: first emapper is homology searches with diamond, second is the actual annotation
```
fastafolder="00_orthology_assignment/proteins_processed/"
outfolder="05_GO/emapper"

cd $fastafolder
for f in *.fasta; do echo "python /apps/eggnogmapper/2.0.0/emapper.py -m diamond --no_annot --no_file_comments --cpu 1 -i 00_orthology_assignment/proteins_processed/$f -o 05_GO/emapper/$f"; done>$outfolder/homology_searches
cd $outfolder

cat homology_searches |slurmtasks -n emapper -m 10 |sbatch

ls |grep emapper.seed_orthologs|cut -f1 -d '.' > species.list
while read species; do echo "emapper.py --annotate_hits_table $species.fasta.emapper.seed_orthologs --no_file_comments -o $species --cpu 2"; done<species_list>annotation_job
cat annotation_job |slurmtasks -n emapper -m 1 -c 2 -f array-2core |sbatch

```
We want to make ids2go file for  all our species. We'll change the comma-delimited GO IDs to semi-colon delimited, as this is the format the GOATOOLS API parses.
```
for species in $(cat species.list); do cut -f 1,7 $species.emapper.annotations|grep GO:|sed 's/,/;/g' > ids2GO_files/$species.ids2go; done
```
Isolate lists of proteins for each species
First, we make lists for all the species of all the proteins in each proteome
```
cd 00_orthology_assignment/proteins_processed/
species_list=$(ls *.fasta|cut -f1 -d '.')
for species in $species_list; do
    grep '>' $species.fasta | cut -f 2 -d '>' > 05_GO/emapper/ids2GO_files/$species.list
done

```
we won't do parahoxozoa enrichments, since blocks we found are roughly as many as background ones 
```

cd 05_GO/GO_enrichment

annotations="../emapper/annotations"
species_list=$(ls $annotations/*emapper.annotations|rev|cut -d '/' -f 1|rev|cut -f1 -d '.') #cut to keep only the last field
ids2go_folder=../emapper/ids2GO_files

#MLCA enrichments
name=Metazoa
syntfile=../../02_REDUX_gene_density_analysis/bynode/CONS/Metazoa.novel.synt
for species in $species_list; do
    grep -P "^\d+\t$species" $syntfile | cut -f10|tr "," "\n"> Metazoa/$species.$name.list;
done

#PLCA enrichments
name=Planulozoa
syntfile=../../02_REDUX_gene_density_analysis/bynode/CONS/Planulozoa.novel.synt
for species in $species_list; do
    grep -P "^\d+\t$species" $syntfile | cut -f10|tr "," "\n"> Planulozoa/$species.$name.list;
done

#BLCA enrichments
name=Bilateria
syntfile=../../02_REDUX_gene_density_analysis/bynode/CONS/Bilateria.novel.synt
for species in $species_list; do
    grep -P "^\d+\t$species" $syntfile | cut -f10|tr "," "\n"> Bilateria/$species.$name.list;
done

#remove empty files (no blocks)
find */* -empty -type f -delete

#get latest version of GO terms (downloaded on the 5th april 2020)
wget http://purl.obolibrary.org/obo/go/go-basic.obo

#run GO enrichment for all species posessing MCLA blocks
species_list=$(ls Metazoa |cut -f2 -d "/"|cut -f1 -d ".") #only species in given folder
for species in $species_list; do
    ids2go=$ids2go_folder/$species.ids2go
    background_seq=$ids2go_folder/$species.list
    sample_seq=Metazoa/$species.Metazoa.list
    python3 /proj/robert/scripts/GO/GO_analysis.py -i $ids2go -go go-basic.obo -b $background_seq -s $sample_seq;
done

#run GO enrichment for all species posessing PCLA blocks
species_list=$(ls Planulozoa |cut -f2 -d "/"|cut -f1 -d ".") #only species in given folder
for species in $species_list; do
    ids2go=$ids2go_folder/$species.ids2go
    background_seq=$ids2go_folder/$species.list
    sample_seq=Planulozoa/$species.Planulozoa.list
    python3 /proj/robert/scripts/GO/GO_analysis.py -i $ids2go -go go-basic.obo -b $background_seq -s $sample_seq;
done

#run GO enrichment for all species posessing BLCA blocks
species_list=$(ls Bilateria |cut -f2 -d "/"|cut -f1 -d ".") #only species in given folder
for species in $species_list; do
    ids2go=$ids2go_folder/$species.ids2go
    background_seq=$ids2go_folder/$species.list
    sample_seq=Bilateria/$species.Bilateria.list
    python3 /proj/robert/scripts/GO/GO_analysis.py -i $ids2go -go go-basic.obo -b $background_seq -s $sample_seq;
done
```

check which terms are enriched in which taxons. GO enrichment results are in the GO_enrichment_results folder
```
cd 05_GO/GO_enrichment
enrichment_comparisons.py
```