## 7. Block correlation
### 7.1 Download Expression data for CALMI and MUSMU
All other transcript abundances taken from Zieger et al. 2020 (doi: 10.1016/j.cub.2020.10.004)
#### CALMI
We first want to use our filtered gff file (no MT genome). And download the transcripts
```
08_exp/CALMI/CALMI.gff

wget ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/165/045/GCF_000165045.1_Callorhinchus_milii-6.1.3/GCF_000165045.1_Callorhinchus_milii-6.1.3_rna.fna.gz -O CALMI.transcripts.fa.gz
```

Now this is for renaming the transcripts in the mRNA file by the protein names and filter out transcripts that are not the ones encoding our filtered proteins
```
/proj/robert/scripts/dev_RNAseq/rename_filter_transcripts.py -a CALMI_NCBI.gff -ft mRNA -fp CDS -kpp Parent -kpt ID -kp protein_id -kt transcript_id -o CALMI_protnames.fna -f CALMI.transcripts.fa -c 01_microsynteny/chrom/CALMI.chrom


```

Download all the reads
```
module load sratoolkit
fasterq-dump SRR1735385 -e 8 -o thymus
fasterq-dump SRR513757 -e 8 -o testis
fasterq-dump SRR513758 -e 8 -o spleen
fasterq-dump SRR513759 -e 8 -o ovary
fasterq-dump SRR513760 -e 8 -o liver
fasterq-dump SRR514104 -e 8 -o muscle
fasterq-dump SRR514105 -e 8 -o kidney
fasterq-dump SRR514106 -e 8 -o intestine
fasterq-dump SRR514107 -e 8 -o heart
fasterq-dump SRR514109 -e 8 -o brain
fasterq-dump SRR534176 -e 8 -o gills
```
Now for the kallisto run
```
module load kallisto
#cerate index
kallisto index -i CALMI.index CALMI_protnames.fna

reads=08_exp/CALMI/reads

le1="_1.fastq"
le2="_2.fastq"

#submit kallisto quantification jobs
for file in $(ls reads/|rev|cut -f 2- -d '_'|rev|uniq); do
    echo "kallisto quant --index=CALMI.index --rf-stranded --output-dir=$file --plaintext reads/$file$le1  reads/$file$le2" ;
done |
/proj/rpz/slurm_scripts/slurmtasks --mem 5 --name quantCALMI |
sbatch
```


```
paste */abundance.tsv |  cut -f 1,5,10,15,20,25,30,35,40,45,50,55,60,65,70,75,80,85,90,95,100,105,110,115 > CALMI_transcript_tpms_all_samples.tsv
ls -1 */abundance.tsv | perl -ne 'chomp $_; if ($_ =~ /(\S+)\/abundance\.tsv/){print "\t$1"}' | perl -ne 'print "target_id$_\n"' > header.tsv
cat header.tsv CALMI_transcript_tpms_all_samples.tsv | grep -v "tpm" > CALMI_transcript_tpms_all_samples.tsv2
mv CALMI_transcript_tpms_all_samples.tsv2 CALMI_transcript_tpms_all_samples.tsv
rm -f header.tsv
```


## MUSMU
```
python3 /proj/robert/scripts/dev_RNAseq/rename_filter_transcripts.py -a 00_orthology_assignment/coordinates/gff/MUSMU_filt_NCBI.gff3 -ft mRNA -fp CDS -kpp gene -kpt gene -kp protein_id -kt Name -o MUSMU_protnames.fna -f MUSMU.transcripts.fa -c 01_microsynteny/chrom/MUSMU.chrom
```

```
fasterq-dump SRR5273648 -e 4 -o female_bone_marrow_a
fasterq-dump SRR5273664 -e 4 -o female_bone_marrow_b


module load sratoolkit
prefetch SRR5273654
cd SRR5273654
fastq-dump --split-e --skip-technical SRR5273654.sra
mv SRR5273654_1.fastq ../female_adrenal_gland_b_1.fastq
mv SRR5273654_2.fastq ../female_adrenal_gland_b_2.fastq

prefetch SRR5273670
cd SRR5273670
fastq-dump --split-e --skip-technical SRR5273670.sra
mv SRR5273670_1.fastq ../female_adrenal_gland_a_1.fastq
mv SRR5273670_2.fastq ../female_adrenal_gland_a_2.fastq
cd ../
rm -r SRR5273670

prefetch SRR5273635
cd SRR5273635
fastq-dump --split-e --skip-technical SRR5273635.sra
mv SRR5273635_1.fastq ../female_brain_a_1.fastq
mv SRR5273635_2.fastq ../female_brain_a_2.fastq
cd ../
rm -r SRR5273635

prefetch SRR5273637
cd SRR5273637
fastq-dump --split-e --skip-technical SRR5273637.sra
mv SRR5273637_1.fastq ../female_brain_b_1.fastq
mv SRR5273637_2.fastq ../female_brain_b_2.fastq
cd ../
rm -r SRR5273637

#todo
prefetch SRR5273657
cd SRR5273657
fastq-dump --split-e --skip-technical SRR5273657.sra
mv SRR5273657_1.fastq ../female_brain_c_1.fastq
mv SRR5273657_2.fastq ../female_brain_c_2.fastq
cd ../
rm -r SRR5273657

prefetch SRR5273673
cd SRR5273673
fastq-dump --split-e --skip-technical SRR5273673.sra
mv SRR5273673_1.fastq ../female_brain_d_1.fastq
mv SRR5273673_2.fastq ../female_brain_d_2.fastq
cd ../
rm -r SRR5273673


prefetch SRR5273646
cd SRR5273646
fastq-dump --split-e --skip-technical SRR5273646.sra
mv SRR5273646_1.fastq ../female_forestomach_a_1.fastq
mv SRR5273646_2.fastq ../female_forestomach_a_2.fastq
cd ../
rm -r SRR5273646


prefetch SRR5273662
cd SRR5273662
fastq-dump --split-e --skip-technical SRR5273662.sra
mv SRR5273662_1.fastq ../female_forestomach_b_1.fastq
mv SRR5273662_2.fastq ../female_forestomach_b_2.fastq
cd ../
rm -r SRR5273662

prefetch SRR5273651
cd SRR5273651
fastq-dump --split-e --skip-technical SRR5273651.sra
mv SRR5273651_1.fastq ../female_heart_a_1.fastq
mv SRR5273651_2.fastq ../female_heart_a_2.fastq
cd ../
rm -r SRR5273651

prefetch SRR5273667
cd SRR5273667
fastq-dump --split-e --skip-technical SRR5273667.sra
mv SRR5273667_1.fastq ../female_heart_b_1.fastq
mv SRR5273667_2.fastq ../female_heart_b_2.fastq
cd ../
rm -r SRR5273667


prefetch SRR5273655
cd SRR5273655
fastq-dump --split-e --skip-technical SRR5273655.sra
mv SRR5273655_1.fastq ../female_kidney_a_1.fastq
mv SRR5273655_2.fastq ../female_kidney_a_2.fastq
cd ../
rm -r SRR5273655

prefetch SRR5273671
cd SRR5273671
fastq-dump --split-e --skip-technical SRR5273671.sra
mv SRR5273671_1.fastq ../female_kidney_b_1.fastq
mv SRR5273671_2.fastq ../female_kidney_b_2.fastq
cd ../
rm -r SRR5273671

prefetch SRR5273644
cd SRR5273644
fastq-dump --split-e --skip-technical SRR5273644.sra
mv SRR5273644_1.fastq ../female_large_intestine_a_1.fastq
mv SRR5273644_2.fastq ../female_large_intestine_a_2.fastq
cd ../
rm -r SRR5273644

prefetch SRR5273660
cd SRR5273660
fastq-dump --split-e --skip-technical SRR5273660.sra
mv SRR5273660_1.fastq ../female_large_intestine_b_1.fastq
mv SRR5273660_2.fastq ../female_large_intestine_b_2.fastq
cd ../
rm -r SRR5273660

prefetch SRR5273634
cd SRR5273634
fastq-dump --split-e --skip-technical SRR5273634.sra
mv SRR5273634_1.fastq ../female_liver_a_1.fastq
mv SRR5273634_2.fastq ../female_liver_a_2.fastq
cd ../
rm -r SRR5273634

prefetch SRR5273636
cd SRR5273636
fastq-dump --split-e --skip-technical SRR5273636.sra
mv SRR5273636_1.fastq ../female_liver_b_1.fastq
mv SRR5273636_2.fastq ../female_liver_b_2.fastq
cd ../
rm -r SRR5273636

prefetch SRR5273656
cd SRR5273656
fastq-dump --split-e --skip-technical SRR5273656.sra
mv SRR5273656_1.fastq ../female_liver_c_1.fastq
mv SRR5273656_2.fastq ../female_liver_c_2.fastq
cd ../
rm -r SRR5273656

prefetch SRR5273672
cd SRR5273672
fastq-dump --split-e --skip-technical SRR5273672.sra
mv SRR5273672_1.fastq ../female_liver_d_1.fastq
mv SRR5273672_2.fastq ../female_liver_d_2.fastq
cd ../
rm -r SRR5273672

prefetch SRR5273652
cd SRR5273652
fastq-dump --split-e --skip-technical SRR5273652.sra
mv SRR5273652_1.fastq ../female_lung_a_1.fastq
mv SRR5273652_2.fastq ../female_lung_a_2.fastq
cd ../
rm -r SRR5273652

prefetch SRR5273668
cd SRR5273668
fastq-dump --split-e --skip-technical SRR5273668.sra
mv SRR5273668_1.fastq ../female_lung_b_1.fastq
mv SRR5273668_2.fastq ../female_lung_b_2.fastq
cd ../
rm -r SRR5273668

prefetch SRR5273643
cd SRR5273643
fastq-dump --split-e --skip-technical SRR5273643.sra
mv SRR5273643_1.fastq ../female_muscle_a_1.fastq
mv SRR5273643_2.fastq ../female_muscle_a_2.fastq
cd ../
rm -r SRR5273643

prefetch SRR5273659
cd SRR5273659
fastq-dump --split-e --skip-technical SRR5273659.sra
mv SRR5273659_1.fastq ../female_muscle_b_1.fastq
mv SRR5273659_2.fastq ../female_muscle_b_2.fastq
cd ../
rm -r SRR5273659

prefetch SRR5273649
cd SRR5273649
fastq-dump --split-e --skip-technical SRR5273649.sra
mv SRR5273649_1.fastq ../female_ovary_a_1.fastq
mv SRR5273649_2.fastq ../female_ovary_a_2.fastq
cd ../
rm -r SRR5273649

prefetch SRR5273665
cd SRR5273665
fastq-dump --split-e --skip-technical SRR5273665.sra
mv SRR5273665_1.fastq ../female_ovary_b_1.fastq
mv SRR5273665_2.fastq ../female_ovary_b_2.fastq
cd ../
rm -r SRR5273665

prefetch SRR5273645
cd SRR5273645
fastq-dump --split-e --skip-technical SRR5273645.sra
mv SRR5273645_1.fastq ../female_small_intestine_a_1.fastq
mv SRR5273645_2.fastq ../female_small_intestine_a_2.fastq
cd ../
rm -r SRR5273645

prefetch SRR5273661
cd SRR5273661
fastq-dump --split-e --skip-technical SRR5273661.sra
mv SRR5273661_1.fastq ../female_small_intestine_b_1.fastq
mv SRR5273661_2.fastq ../female_small_intestine_b_2.fastq
cd ../
rm -r SRR5273661

prefetch SRR5273653
cd SRR5273653
fastq-dump --split-e --skip-technical SRR5273653.sra
mv SRR5273653_1.fastq ../female_spleen_a_1.fastq
mv SRR5273653_2.fastq ../female_spleen_a_2.fastq
cd ../
rm -r SRR5273653

prefetch SRR5273669
cd SRR5273669
fastq-dump --split-e --skip-technical SRR5273669.sra
mv SRR5273669_1.fastq ../female_spleen_b_1.fastq
mv SRR5273669_2.fastq ../female_spleen_b_2.fastq
cd ../
rm -r SRR5273669

prefetch SRR5273647
cd SRR5273647
fastq-dump --split-e --skip-technical SRR5273647.sra
mv SRR5273647_1.fastq ../female_stomach_a_1.fastq
mv SRR5273647_2.fastq ../female_stomach_a_2.fastq
cd ../
rm -r SRR5273647

prefetch SRR5273663
cd SRR5273663
fastq-dump --split-e --skip-technical SRR5273663.sra
mv SRR5273663_1.fastq ../female_stomach_b_1.fastq
mv SRR5273663_2.fastq ../female_stomach_b_2.fastq
cd ../
rm -r SRR5273663

prefetch SRR5273650
cd SRR5273650
fastq-dump --split-e --skip-technical SRR5273650.sra
mv SRR5273650_1.fastq ../female_thymus_a_1.fastq
mv SRR5273650_2.fastq ../female_thymus_a_2.fastq
cd ../
rm -r SRR5273650

prefetch SRR5273666
cd SRR5273666
fastq-dump --split-e --skip-technical SRR5273666.sra
mv SRR5273666_1.fastq ../female_thymus_b_1.fastq
mv SRR5273666_2.fastq ../female_thymus_b_2.fastq
cd ../
rm -r SRR5273666

prefetch SRR5273642
cd SRR5273642
fastq-dump --split-e --skip-technical SRR5273642.sra
mv SRR5273642_1.fastq ../female_uterus_1.fastq
mv SRR5273642_2.fastq ../female_uterus_2.fastq
cd ../
rm -r SRR5273642

prefetch SRR5273658
cd SRR5273658
fastq-dump --split-e --skip-technical SRR5273658.sra
mv SRR5273658_1.fastq ../female_vesicular_gland_1.fastq
mv SRR5273658_2.fastq ../female_vesicular_gland_2.fastq
cd ../
rm -r SRR5273658

prefetch SRR5273686
cd SRR5273686
fastq-dump --split-e --skip-technical SRR5273686.sra
mv SRR5273686_1.fastq ../male_adrenal_gland_a_1.fastq
mv SRR5273686_2.fastq ../male_adrenal_gland_a_2.fastq
cd ../
rm -r SRR5273686

prefetch SRR5273702
cd SRR5273702
fastq-dump --split-e --skip-technical SRR5273702.sra
mv SRR5273702_1.fastq ../male_adrenal_gland_b_1.fastq
mv SRR5273702_2.fastq ../male_adrenal_gland_b_2.fastq
cd ../
rm -r SRR5273702

prefetch SRR5273680
cd SRR5273680
fastq-dump --split-e --skip-technical SRR5273680.sra
mv SRR5273680_1.fastq ../male_bone_marrow_a_1.fastq
mv SRR5273680_2.fastq ../male_bone_marrow_a_2.fastq
cd ../
rm -r SRR5273680

prefetch SRR5273696
cd SRR5273696
fastq-dump --split-e --skip-technical SRR5273696.sra
mv SRR5273696_1.fastq ../male_bone_marrow_b_1.fastq
mv SRR5273696_2.fastq ../male_bone_marrow_b_2.fastq
cd ../
rm -r SRR5273696

prefetch SRR5273639
cd SRR5273639
fastq-dump --split-e --skip-technical SRR5273639.sra
mv SRR5273639_1.fastq ../male_brain_a_1.fastq
mv SRR5273639_2.fastq ../male_brain_a_2.fastq
cd ../
rm -r SRR5273639

prefetch SRR5273641
cd SRR5273641
fastq-dump --split-e --skip-technical SRR5273641.sra
mv SRR5273641_1.fastq ../male_brain_b_1.fastq
mv SRR5273641_2.fastq ../male_brain_b_2.fastq
cd ../
rm -r SRR5273641

prefetch SRR5273689
cd SRR5273689
fastq-dump --split-e --skip-technical SRR5273689.sra
mv SRR5273689_1.fastq ../male_brain_c_1.fastq
mv SRR5273689_2.fastq ../male_brain_c_2.fastq
cd ../
rm -r SRR5273689

prefetch SRR5273705
cd SRR5273705
fastq-dump --split-e --skip-technical SRR5273705.sra
mv SRR5273705_1.fastq ../male_brain_d_1.fastq
mv SRR5273705_2.fastq ../male_brain_d_2.fastq
cd ../
rm -r SRR5273705

prefetch SRR5273678
cd SRR5273678
fastq-dump --split-e --skip-technical SRR5273678.sra
mv SRR5273678_1.fastq ../male_forestomach_a_1.fastq
mv SRR5273678_2.fastq ../male_forestomach_a_2.fastq
cd ../
rm -r SRR5273678

prefetch SRR5273694
cd SRR5273694
fastq-dump --split-e --skip-technical SRR5273694.sra
mv SRR5273694_1.fastq ../male_forestomach_b_1.fastq
mv SRR5273694_2.fastq ../male_forestomach_b_2.fastq
cd ../
rm -r SRR5273694

prefetch SRR5273683
cd SRR5273683
fastq-dump --split-e --skip-technical SRR5273683.sra
mv SRR5273683_1.fastq ../male_heart_a_1.fastq
mv SRR5273683_2.fastq ../male_heart_a_2.fastq
cd ../
rm -r SRR5273683

prefetch SRR5273699
cd SRR5273699
fastq-dump --split-e --skip-technical SRR5273699.sra
mv SRR5273699_1.fastq ../male_heart_b_1.fastq
mv SRR5273699_2.fastq ../male_heart_b_2.fastq
cd ../
rm -r SRR5273699

prefetch SRR5273687
cd SRR5273687
fastq-dump --split-e --skip-technical SRR5273687.sra
mv SRR5273687_1.fastq ../male_kidney_a_1.fastq
mv SRR5273687_2.fastq ../male_kidney_a_2.fastq
cd ../
rm -r SRR5273687

prefetch SRR5273703
cd SRR5273703
fastq-dump --split-e --skip-technical SRR5273703.sra
mv SRR5273703_1.fastq ../male_kidney_b_1.fastq
mv SRR5273703_2.fastq ../male_kidney_b_2.fastq
cd ../
rm -r SRR5273703

prefetch SRR5273676
cd SRR5273676
fastq-dump --split-e --skip-technical SRR5273676.sra
mv SRR5273676_1.fastq ../male_large_intestine_a_1.fastq
mv SRR5273676_2.fastq ../male_large_intestine_a_2.fastq
cd ../
rm -r SRR5273676

prefetch SRR5273692
cd SRR5273692
fastq-dump --split-e --skip-technical SRR5273692.sra
mv SRR5273692_1.fastq ../male_large_intestine_b_1.fastq
mv SRR5273692_2.fastq ../male_large_intestine_b_2.fastq
cd ../
rm -r SRR5273692

prefetch SRR5273638
cd SRR5273638
fastq-dump --split-e --skip-technical SRR5273638.sra
mv SRR5273638_1.fastq ../male_liver_a_1.fastq
mv SRR5273638_2.fastq ../male_liver_a_2.fastq
cd ../
rm -r SRR5273638

prefetch SRR5273640
cd SRR5273640
fastq-dump --split-e --skip-technical SRR5273640.sra
mv SRR5273640_1.fastq ../male_liver_b_1.fastq
mv SRR5273640_2.fastq ../male_liver_b_2.fastq
cd ../
rm -r SRR5273640

prefetch SRR5273688
cd SRR5273688
fastq-dump --split-e --skip-technical SRR5273688.sra
mv SRR5273688_1.fastq ../male_liver_c_1.fastq
mv SRR5273688_2.fastq ../male_liver_c_2.fastq
cd ../
rm -r SRR5273688

prefetch SRR5273704
cd SRR5273704
fastq-dump --split-e --skip-technical SRR5273704.sra
mv SRR5273704_1.fastq ../male_liver_d_1.fastq
mv SRR5273704_2.fastq ../male_liver_d_2.fastq
cd ../
rm -r SRR5273704

prefetch SRR5273684
cd SRR5273684
fastq-dump --split-e --skip-technical SRR5273684.sra
mv SRR5273684_1.fastq ../male_lung_a_1.fastq
mv SRR5273684_2.fastq ../male_lung_a_2.fastq
cd ../
rm -r SRR5273684

prefetch SRR5273700
cd SRR5273700
fastq-dump --split-e --skip-technical SRR5273700.sra
mv SRR5273700_1.fastq ../male_lung_b_1.fastq
mv SRR5273700_2.fastq ../male_lung_b_2.fastq
cd ../
rm -r SRR5273700

prefetch SRR5273675
cd SRR5273675
fastq-dump --split-e --skip-technical SRR5273675.sra
mv SRR5273675_1.fastq ../male_muscle_a_1.fastq
mv SRR5273675_2.fastq ../male_muscle_a_2.fastq
cd ../
rm -r SRR5273675

prefetch SRR5273691
cd SRR5273691
fastq-dump --split-e --skip-technical SRR5273691.sra
mv SRR5273691_1.fastq ../male_muscle_b_1.fastq
mv SRR5273691_2.fastq ../male_muscle_b_2.fastq
cd ../
rm -r SRR5273691

prefetch SRR5273677
cd SRR5273677
fastq-dump --split-e --skip-technical SRR5273677.sra
mv SRR5273677_1.fastq ../male_small_intestine_a_1.fastq
mv SRR5273677_2.fastq ../male_small_intestine_a_2.fastq
cd ../
rm -r SRR5273677

prefetch SRR5273693
cd SRR5273693
fastq-dump --split-e --skip-technical SRR5273693.sra
mv SRR5273693_1.fastq ../male_small_intestine_b_1.fastq
mv SRR5273693_2.fastq ../male_small_intestine_b_2.fastq
cd ../
rm -r SRR5273693

prefetch SRR5273685
cd SRR5273685
fastq-dump --split-e --skip-technical SRR5273685.sra
mv SRR5273685_1.fastq ../male_spleen_a_1.fastq
mv SRR5273685_2.fastq ../male_spleen_a_2.fastq
cd ../
rm -r SRR5273685

prefetch SRR5273701
cd SRR5273701
fastq-dump --split-e --skip-technical SRR5273701.sra
mv SRR5273701_1.fastq ../male_spleen_b_1.fastq
mv SRR5273701_2.fastq ../male_spleen_b_2.fastq
cd ../
rm -r SRR5273701

prefetch SRR5273679
cd SRR5273679
fastq-dump --split-e --skip-technical SRR5273679.sra
mv SRR5273679_1.fastq ../male_stomach_a_1.fastq
mv SRR5273679_2.fastq ../male_stomach_a_2.fastq
cd ../
rm -r SRR5273679

prefetch SRR5273695
cd SRR5273695
fastq-dump --split-e --skip-technical SRR5273695.sra
mv SRR5273695_1.fastq ../male_stomach_b_1.fastq
mv SRR5273695_2.fastq ../male_stomach_b_2.fastq
cd ../
rm -r SRR5273695

prefetch SRR5273681
cd SRR5273681
fastq-dump --split-e --skip-technical SRR5273681.sra
mv SRR5273681_1.fastq ../male_testis_a_1.fastq
mv SRR5273681_2.fastq ../male_testis_a_2.fastq
cd ../
rm -r SRR5273681

prefetch SRR5273697
cd SRR5273697
fastq-dump --split-e --skip-technical SRR5273697.sra
mv SRR5273697_1.fastq ../male_testis_b_1.fastq
mv SRR5273697_2.fastq ../male_testis_b_2.fastq
cd ../
rm -r SRR5273697

prefetch SRR5273682
cd SRR5273682
fastq-dump --split-e --skip-technical SRR5273682.sra
mv SRR5273682_1.fastq ../male_thymus_a_1.fastq
mv SRR5273682_2.fastq ../male_thymus_a_2.fastq
cd ../
rm -r SRR5273682

prefetch SRR5273698
cd SRR5273698
fastq-dump --split-e --skip-technical SRR5273698.sra
mv SRR5273698_1.fastq ../male_thymus_b_1.fastq
mv SRR5273698_2.fastq ../male_thymus_b_2.fastq
cd ../
rm -r SRR5273698

prefetch SRR5273674
cd SRR5273674
fastq-dump --split-e --skip-technical SRR5273674.sra
mv SRR5273674_1.fastq ../male_vesicular_gland_a_1.fastq
mv SRR5273674_2.fastq ../male_vesicular_gland_a_2.fastq
cd ../
rm -r SRR5273674

prefetch SRR5273690
cd SRR5273690
fastq-dump --split-e --skip-technical SRR5273690.sra
mv SRR5273690_1.fastq ../male_vesicular_gland_b_1.fastq
mv SRR5273690_2.fastq ../male_vesicular_gland_b_2.fastq
cd ../
rm -r SRR5273690
```
kallisto run
```
module load kallisto
#cerate index
kallisto index -i MUSMU.index MUSMU_protnames.fna

reads=08_exp/MUSMU/reads

le1="_1.fastq"
le2="_2.fastq"

#submit kallisto quantification jobs
for file in $(ls reads/|rev|cut -f 2- -d '_'|rev|uniq); do
    echo "kallisto quant --index=MUSMU.index --rf-stranded --output-dir=$file --plaintext reads/$file$le1  reads/$file$le2" ;
done |
/proj/rpz/slurm_scripts/slurmtasks --mem 5 --name quantMUSMU |
sbatch



paste */abundance.tsv |  cut -f 1,5,10,15,20,25,30,35,40,45,50,55,60,65,70,75,80,85,90,95,100,105,110,115,120,125,130,135,140,145,150,155,160,165,170,175,180,185,190,195,200,205,210,215,220,225,230,235,240,245,250,255,260,265,270,275,280,285,290,295,300,305,310,315,320,325,330,335,340,345,350,355,360,365,370,375,380,385,390 > MUSMU_transcript_tpms_all_samples.tsv
ls -1 */abundance.tsv | perl -ne 'chomp $_; if ($_ =~ /(\S+)\/abundance\.tsv/){print "\t$1"}' | perl -ne 'print "target_id$_\n"' > header.tsv
cat header.tsv MUSMU_transcript_tpms_all_samples.tsv | grep -v "tpm" > MUSMU_transcript_tpms_all_samples.tsv2
mv MUSMU_transcript_tpms_all_samples.tsv2 MUSMU_transcript_tpms_all_samples.tsv
rm -f header.tsv

python3 /proj/robert/scripts/Utilities/make_tpm_stage_medians.py "female_adrenal_gland female_bone_marrow female_brain female_forestomach female_heart female_kidney female_large_intestine female_liver female_lung female_muscle female_ovary female_small_intestine female_spleen female_stomach female_thymus female_uterus female_vesicular_gland male_adrenal_gland male_bone_marrow male_brain male_forestomach male_heart male_kidney male_large_intestine male_liver male_lung male_muscle male_small_intestine male_spleen male_stomach male_testis male_thymus male_vesicular_gland" -i MUSMU_transcript_tpms_all_samples.tsv

```

## Calculate the actual block correlations
Really large blocks in vertebrates. since it involve a bit of combinations, it can take a while, especially with the long gnathostome blocks.

Elephant shark
```
python3 /proj/robert/scripts/density/block_correlation_analysis.py -t 02_gene_density_analysis/density_whole_genome/key_nodes.tidydf.csv -s 01_microsynteny/chrom_of/5.blocks.3.syn_corrected.synt -e CALMI/CALMI_transcript_tpms_all_samples.tsv -p CALMI -o CALMI_tidy_corr.csv 
```
Scallop
```
python3 /proj/robert/scripts/density/block_correlation_analysis.py -t 02_gene_density_analysis/density_whole_genome/key_nodes.tidydf.csv -s 01_microsynteny/chrom_of/5.blocks.3.syn_corrected.synt -e MIZYE/MIZYE_transcript_tpms_all_samples_medians.tsv -p MIZYE -o MIZYE_tidy_corr.csv 
```
Mouse
```
python3 /proj/robert/scripts/density/block_correlation_analysis.py -t 02_gene_density_analysis/density_whole_genome/key_nodes.tidydf.csv -s 01_microsynteny/chrom_of/5.blocks.3.syn_corrected.synt -e MUSMU/MUSMU_transcript_tpms_all_samples_medians.tsv -p MUSMU -o MUSMU_tidy_corr.csv 
```
Oyster
```
python3 /proj/robert/scripts/density/block_correlation_analysis.py -t 02_gene_density_analysis/density_whole_genome/key_nodes.tidydf.csv -s 01_microsynteny/chrom_of/5.blocks.3.syn_corrected.synt -e CRAGI/CRAGI_transcript_tpms_all_samples.tsv -p CRAGI -o CRAGI_tidy_corr.csv 
```
Urchin
```
python3 /proj/robert/scripts/density/block_correlation_analysis.py -t 02_gene_density_analysis/density_whole_genome/key_nodes.tidydf.csv -s 01_microsynteny/chrom_of/5.blocks.3.syn_corrected.synt -e STRPU/STRPU_transcript_tpms_all_samples.tsv -p STRPU -o STRPU_tidy_corr.csv 
```
Hemichordate
```
python3 /proj/robert/scripts/density/block_correlation_analysis.py -t 02_gene_density_analysis/density_whole_genome/key_nodes.tidydf.csv -s 01_microsynteny/chrom_of/5.blocks.3.syn_corrected.synt -e SACKO/SACKO_transcript_tpms_all_samples.tsv -p SACKO -o SACKO_tidy_corr.csv 
```
