## 8. Particular cases manual annotation
### 8.1 HOX
#### 8.1.1 HOX annotation
References and animals for which *hox* sequences used as queries for reciprocal BLAST searches in the planulozoan sequences of our sample:

* Amemiya et al. 2013 (doi: doi.org/10.1038/nature12027); LATCH

* Anaya et al. 2013 (doi:10.1186/1471-213X-13-26); HOMSA, MUSMU, DANRE, CALMI, CHEMY, CIOIN, BRALA, STRPU, SACKO, PTYFL

* Belcaid et al. 2018 (doi: 10.1073/pnas.1817322116); EUPSC

* Brauchle et al. 2018 (doi: doi.org/10.1093/gbe/evy170); HOFMI

* Currie et al. 2016 (doi: 10.1186/s13227-016-0044-8); SCHME

* DuBuc et al. 2012 (doi: 10.1093/icb/ics098); NEMVE

* Hench et al. 2015 (doi: 10.1371/journal.pone.0126947); CAEEL

* Kuraku and Meyer 2009 (doi: 10.1387/ijdb.072533km); DANRE, HOMSA, GALGA, XENTR, CALMI

* LeclÃ¨re et al. 2019 (doi: 10.1038/s41559-019-0833-2); CLYHE

* Leite et al. 2018 (doi: 10.1093/molbev/msy125); PARTE

* Pace et al. 2016 (doi: 10.1186/s13227-016-0048-4); ANOGA, DROME, TRICA, DAPPU, STRMA, IXOSC

* Simakov et al. 2013 (doi: 10.1038/nature11696); CAPTE, LOTGI, DAPPU

* Simakov et al. 2015 (doi: 10.1038/nature16150); PTYFL, SACKO

* Wang et al. 2017 (doi: 10.1038/s41559-017-0120); MIZYE

Orthologs of hoxes in the 49 species of our sample were identified by reciprocal BLASTs using as queries aforementionned sequences.
Annotation was given according to their top hit in the NR database.

#### 8.1.2 Hox microsynteny and gene density analysis
`HOX.clus` Is a cluster file constituted of only one OG (accessions of all annoatted hoxes in http://synteny.csb.univie.ac.at/, planulozoan novelty, `hox manually curated`).

```
cd 01_microsynteny/chrom
prepMicroSynt.pl ACAPL.chrom,ACRMI.chrom,ADIVA.chrom,AMPQU.chrom,ANOGA.chrom,AURAU.chrom,BRALA.chrom,CAEEL.chrom,CALMI.chrom,CAPOW.chrom,CAPTE.chrom,CHEMY.chrom,CIOIN.chrom,CLYHE.chrom,CRAGI.chrom,DANRE.chrom,DAPPU.chrom,DROME.chrom,EUPSC.chrom,EXAPA.chrom,GALGA.chrom,HELRO.chrom,HIPCO.chrom,HOFMI.chrom,HOIHO.chrom,HOMSA.chrom,HYDVU.chrom,IXOSC.chrom,LATCH.chrom,LEPOC.chrom,LINAN.chrom,LOTGI.chrom,MAYZE.chrom,MIZYE.chrom,MNELE.chrom,MUSMU.chrom,NEMVE.chrom,PARTE.chrom,PLEBA.chrom,PTYFL.chrom,SACKO.chrom,SALRO.chrom,SCHME.chrom,STRMA.chrom,STRPU.chrom,SYCCI.chrom,TRIAD.chrom,TRICA.chrom,XENTR.chrom 5 03_HOX/HOX_density/HOX.clus
sbatch --array=1-1176 --constraint=array-1core --job-name=hox_synt job.sh

chromfile_folder='01_microsynteny/chrom'
makeClusters3.pl $chromfile_folder/ACAPL.chrom,$chromfile_folder/ACRMI.chrom,$chromfile_folder/ADIVA.chrom,$chromfile_folder/AMPQU.chrom,$chromfile_folder/ANOGA.chrom,$chromfile_folder/AURAU.chrom,$chromfile_folder/BRALA.chrom,$chromfile_folder/CAEEL.chrom,$chromfile_folder/CALMI.chrom,$chromfile_folder/CAPOW.chrom,$chromfile_folder/CAPTE.chrom,$chromfile_folder/CHEMY.chrom,$chromfile_folder/CIOIN.chrom,$chromfile_folder/CLYHE.chrom,$chromfile_folder/CRAGI.chrom,$chromfile_folder/DANRE.chrom,$chromfile_folder/DAPPU.chrom,$chromfile_folder/DROME.chrom,$chromfile_folder/EUPSC.chrom,$chromfile_folder/EXAPA.chrom,$chromfile_folder/GALGA.chrom,$chromfile_folder/HELRO.chrom,$chromfile_folder/HIPCO.chrom,$chromfile_folder/HOFMI.chrom,$chromfile_folder/HOIHO.chrom,$chromfile_folder/HOMSA.chrom,$chromfile_folder/HYDVU.chrom,$chromfile_folder/IXOSC.chrom,$chromfile_folder/LATCH.chrom,$chromfile_folder/LEPOC.chrom,$chromfile_folder/LINAN.chrom,$chromfile_folder/LOTGI.chrom,$chromfile_folder/MAYZE.chrom,$chromfile_folder/MIZYE.chrom,$chromfile_folder/MNELE.chrom,$chromfile_folder/MUSMU.chrom,$chromfile_folder/NEMVE.chrom,$chromfile_folder/PARTE.chrom,$chromfile_folder/PLEBA.chrom,$chromfile_folder/PTYFL.chrom,$chromfile_folder/SACKO.chrom,$chromfile_folder/SALRO.chrom,$chromfile_folder/SCHME.chrom,$chromfile_folder/STRMA.chrom,$chromfile_folder/STRPU.chrom,$chromfile_folder/SYCCI.chrom,$chromfile_folder/TRIAD.chrom,$chromfile_folder/TRICA.chrom,$chromfile_folder/XENTR.chrom .5.blocks 3 0.3 0.5  03_HOX/HOX_density/5.blocks.3.syn.synt

correct_blocks_coordinates.py 03_HOX/HOX_density/5.blocks.3.syn.synt ACAPL.chrom,ACRMI.chrom,ADIVA.chrom,AMPQU.chrom,ANOGA.chrom,AURAU.chrom,BRALA.chrom,CAEEL.chrom,CALMI.chrom,CAPOW.chrom,CAPTE.chrom,CHEMY.chrom,CIOIN.chrom,CLYHE.chrom,CRAGI.chrom,DANRE.chrom,DAPPU.chrom,DROME.chrom,EUPSC.chrom,EXAPA.chrom,GALGA.chrom,HELRO.chrom,HIPCO.chrom,HOFMI.chrom,HOIHO.chrom,HOMSA.chrom,HYDVU.chrom,IXOSC.chrom,LATCH.chrom,LEPOC.chrom,LINAN.chrom,LOTGI.chrom,MAYZE.chrom,MIZYE.chrom,MNELE.chrom,MUSMU.chrom,NEMVE.chrom,PARTE.chrom,PLEBA.chrom,PTYFL.chrom,SACKO.chrom,SALRO.chrom,SCHME.chrom,STRMA.chrom,STRPU.chrom,SYCCI.chrom,TRIAD.chrom,TRICA.chrom,XENTR.chrom > 03_HOX/HOX_density/5.blocks.3.syn_corrected.synt

```
Make the randomized HOX blocks.
```
cd 03_HOX/HOX_density/rand_blocks
cut -f2 ../5.blocks.3.syn_corrected.synt|sort -u > genomes.list
n=100
outfolder='03_HOX/HOX_density/rand_blocks'
chromfile_folder='01_microsynteny/chrom'
inputfile='../5.blocks.3.syn_corrected.synt'

while read species; do echo "pick_random_blocks.R --only-genome=$species -n $n $outfolder $inputfile $chromfile_folder/$species.chrom"; done < genomes.list > random_hoxblocks

slurmtasks -m 10 -n random_hoxblocks random_hoxblocks | sbatch
cat *.tsv | grep -v block_id > Hoxblocks.100r.syntx #merge all the tsvs together

cd $chromfile_folder
correct_blocks_coordinates.py 03_HOX/HOX_density/rand_blocks/Hoxblocks.100r.syntx ACAPL.chrom,ACRMI.chrom,ADIVA.chrom,AMPQU.chrom,ANOGA.chrom,AURAU.chrom,BRALA.chrom,CAEEL.chrom,CALMI.chrom,CAPOW.chrom,CAPTE.chrom,CHEMY.chrom,CIOIN.chrom,CLYHE.chrom,CRAGI.chrom,DANRE.chrom,DAPPU.chrom,DROME.chrom,EUPSC.chrom,EXAPA.chrom,GALGA.chrom,HELRO.chrom,HIPCO.chrom,HOFMI.chrom,HOIHO.chrom,HOMSA.chrom,HYDVU.chrom,IXOSC.chrom,LATCH.chrom,LEPOC.chrom,LINAN.chrom,LOTGI.chrom,MAYZE.chrom,MIZYE.chrom,MNELE.chrom,MUSMU.chrom,NEMVE.chrom,PARTE.chrom,PLEBA.chrom,PTYFL.chrom,SACKO.chrom,SALRO.chrom,SCHME.chrom,STRMA.chrom,STRPU.chrom,SYCCI.chrom,TRIAD.chrom,TRICA.chrom,XENTR.chrom > 03_HOX/HOX_density/rand_blocks/Hoxblocks.100r.synt

```

```
python3 make_tidy_density_df.py -s ../../03_REDUX_density_HOX_WNT/hox -g ../../02_REDUX_gene_density_analysis/genomes/ -c ../../01_microsynteny/chrom/ -m ../../03_HOX/hox_graph_data/hox_planu.clus -og ../..//03_HOX/hox_graph_data/hox_planu.clus -o hox
```


### 8.2 WNT
#### 8.2.1 Characterization of wnt5-wnt7 cluster
The cluster detected by the microsynteny pipeline is cluster 953, planulozoan novelty (http://synteny.csb.univie.ac.at/).

References for evidence of synteny of members of the wnt5-wnt7 cluster:

* Cho et al. 2011 (doi: 10.1093/molbev/msq052)

* Janssen et al. 2010 (doi: 10.1186/1471-2148-10-374)

* Sullivan et al. 2007 (doi: 10.1007/s00427-007-0136-5)

* Irimia et al. 2012 (doi: 10.1101/gr.139725.112)

* Kapasa et al. 2010 (doi: 10.1186/1745-6150-5-49)

* Garriock et al. 2012 (doi: 10.1002/dvdy.21156)

Known syntenies by reference

* *wnt5* and *wnt7* in Planulozoans: (Cho et al. 2011), (Janssen et al. 2010), (Sullivan et al. 2007)

* *wnt5* and *fbxl14* in Planulozoans (Irimia et al. 2012), and bilaterians (Kapasa et al. 2010)

* *wnt7* and *atxn10* in Planulozoans (Irimia et al. 2012) in Vertebrates (Garriock et al. 2012)

* *wnt5* and *erc1/2* in Olfactores (Kapasa at al. 2010)

* *wnt5* and *cacna2d* in Vertebrates (Kapasa et al. 2010, Garriock et al. 2012)

* *cacna1d*, *ninj1/2* and *dcp1*, this study


Orthologs to members of the *wnt5-wnt7* cluster (*wnt5-wnt7* pair, *fbxl14*, *atxn10*, *erc1/2*, *cacna1d*, *cacna2d* *ninj1/ninj2* and *dcp1a/dcp1b*) in the planulozoans of the 49 species of our sample were identified by reciprocal BLAST using as inital queries human sequences.
#### 8.2.2 wnt5-wnt7 microsynteny and gene density analysis

Similarmy to what we did with hoxes, we build an orthology file where all members of the cluster are in a single OG (*wnt5-wnt7* pair, *fbxl14*, *atxn10*, *erc1/2*, *cacna1d*, *cacna2d* *ninj1/ninj2* and *dcp1a/dcp1b*), the file is `wnt5_wnt7.clus`

```
cd 01_microsynteny/chrom
perl /proj/Simakov/scripts/MICROSYNT/prepMicroSynt.pl ACAPL.chrom,ACRMI.chrom,ADIVA.chrom,ANOGA.chrom,AURAU.chrom,BRALA.chrom,CAEEL.chrom,CALMI.chrom,CAPTE.chrom,CHEMY.chrom,CIOIN.chrom,CLYHE.chrom,CRAGI.chrom,DANRE.chrom,DAPPU.chrom,DROME.chrom,EUPSC.chrom,EXAPA.chrom,GALGA.chrom,HELRO.chrom,HIPCO.chrom,HOFMI.chrom,HOMSA.chrom,HYDVU.chrom,IXOSC.chrom,LATCH.chrom,LEPOC.chrom,LINAN.chrom,LOTGI.chrom,MAYZE.chrom,MIZYE.chrom,MUSMU.chrom,NEMVE.chrom,PARTE.chrom,PTYFL.chrom,SACKO.chrom,SCHME.chrom,STRMA.chrom,STRPU.chrom,TRICA.chrom,XENTR.chrom 5 07_wnt5_wnt7/wnt5_wnt7.clus
sbatch --array=1-820 --constraint=array-1core --job-name=wnt5_wnt7 job.sh

chromfile_folder='01_microsynteny/chrom'
perl /proj/Simakov/scripts/MICROSYNT/makeClusters3.pl $chromfile_folder/ACAPL.chrom,$chromfile_folder/ACRMI.chrom,$chromfile_folder/ADIVA.chrom,$chromfile_folder/AMPQU.chrom,$chromfile_folder/ANOGA.chrom,$chromfile_folder/AURAU.chrom,$chromfile_folder/BRALA.chrom,$chromfile_folder/CAEEL.chrom,$chromfile_folder/CALMI.chrom,$chromfile_folder/CAPOW.chrom,$chromfile_folder/CAPTE.chrom,$chromfile_folder/CHEMY.chrom,$chromfile_folder/CIOIN.chrom,$chromfile_folder/CLYHE.chrom,$chromfile_folder/CRAGI.chrom,$chromfile_folder/DANRE.chrom,$chromfile_folder/DAPPU.chrom,$chromfile_folder/DROME.chrom,$chromfile_folder/EUPSC.chrom,$chromfile_folder/EXAPA.chrom,$chromfile_folder/GALGA.chrom,$chromfile_folder/HELRO.chrom,$chromfile_folder/HIPCO.chrom,$chromfile_folder/HOFMI.chrom,$chromfile_folder/HOIHO.chrom,$chromfile_folder/HOMSA.chrom,$chromfile_folder/HYDVU.chrom,$chromfile_folder/IXOSC.chrom,$chromfile_folder/LATCH.chrom,$chromfile_folder/LEPOC.chrom,$chromfile_folder/LINAN.chrom,$chromfile_folder/LOTGI.chrom,$chromfile_folder/MAYZE.chrom,$chromfile_folder/MIZYE.chrom,$chromfile_folder/MNELE.chrom,$chromfile_folder/MUSMU.chrom,$chromfile_folder/NEMVE.chrom,$chromfile_folder/PARTE.chrom,$chromfile_folder/PLEBA.chrom,$chromfile_folder/PTYFL.chrom,$chromfile_folder/SACKO.chrom,$chromfile_folder/SALRO.chrom,$chromfile_folder/SCHME.chrom,$chromfile_folder/STRMA.chrom,$chromfile_folder/STRPU.chrom,$chromfile_folder/SYCCI.chrom,$chromfile_folder/TRIAD.chrom,$chromfile_folder/TRICA.chrom,$chromfile_folder/XENTR.chrom .5.blocks 3 0.3 0.5 > 07_wnt5_wnt7/wnt5_wnt7_density/pairwise_blocks/5.blocks.3.syn.synt

cd $chromfile_folder
correct_blocks_coordinates.py 07_wnt5_wnt7/wnt5_wnt7_density/5.blocks.3.syn.synt ACAPL.chrom,ACRMI.chrom,ADIVA.chrom,AMPQU.chrom,ANOGA.chrom,AURAU.chrom,BRALA.chrom,CAEEL.chrom,CALMI.chrom,CAPOW.chrom,CAPTE.chrom,CHEMY.chrom,CIOIN.chrom,CLYHE.chrom,CRAGI.chrom,DANRE.chrom,DAPPU.chrom,DROME.chrom,EUPSC.chrom,EXAPA.chrom,GALGA.chrom,HELRO.chrom,HIPCO.chrom,HOFMI.chrom,HOIHO.chrom,HOMSA.chrom,HYDVU.chrom,IXOSC.chrom,LATCH.chrom,LEPOC.chrom,LINAN.chrom,LOTGI.chrom,MAYZE.chrom,MIZYE.chrom,MNELE.chrom,MUSMU.chrom,NEMVE.chrom,PARTE.chrom,PLEBA.chrom,PTYFL.chrom,SACKO.chrom,SALRO.chrom,SCHME.chrom,STRMA.chrom,STRPU.chrom,SYCCI.chrom,TRIAD.chrom,TRICA.chrom,XENTR.chrom > 07_wnt5_wnt7/wnt5_wnt7_density/5.blocks.3.syn_corrected.synt

cd 07_wnt5_wnt7/wnt5_wnt7_density/rand_blocks

cut -f2 ../5.blocks.3.syn_corrected.synt|sort -u > genomes.list
n=100
outfolder='07_wnt5_wnt7/wnt5_wnt7_density/rand_blocks'
chromfile_folder='01_microsynteny/chrom'
inputfile='../5.blocks.3.syn_corrected.synt'

while read species; do echo "Rscript  /proj/robert/scripts/microsynteny/pick_random_blocks.R --only-genome=$species -n $n $outfolder $inputfile $chromfile_folder/$species.chrom"; done < genomes.list > wnt5_wnt7

nohup bash wnt5_wnt7&

cat *.tsv | grep -v block_id > wnt5_wnt7.100r.syntx #merge all the tsvs together

cd $chromfile_folder
correct_blocks_coordinates.py 07_wnt5_wnt7/wnt5_wnt7_density/rand_blocks/wnt5_wnt7.100r.syntx ACAPL.chrom,ACRMI.chrom,ADIVA.chrom,AMPQU.chrom,ANOGA.chrom,AURAU.chrom,BRALA.chrom,CAEEL.chrom,CALMI.chrom,CAPOW.chrom,CAPTE.chrom,CHEMY.chrom,CIOIN.chrom,CLYHE.chrom,CRAGI.chrom,DANRE.chrom,DAPPU.chrom,DROME.chrom,EUPSC.chrom,EXAPA.chrom,GALGA.chrom,HELRO.chrom,HIPCO.chrom,HOFMI.chrom,HOIHO.chrom,HOMSA.chrom,HYDVU.chrom,IXOSC.chrom,LATCH.chrom,LEPOC.chrom,LINAN.chrom,LOTGI.chrom,MAYZE.chrom,MIZYE.chrom,MNELE.chrom,MUSMU.chrom,NEMVE.chrom,PARTE.chrom,PLEBA.chrom,PTYFL.chrom,SACKO.chrom,SALRO.chrom,SCHME.chrom,STRMA.chrom,STRPU.chrom,SYCCI.chrom,TRIAD.chrom,TRICA.chrom,XENTR.chrom > 07_wnt5_wnt7/wnt5_wnt7_density/rand_blocks/wnt5_wnt7.100r.synt


python3 make_tidy_density_df.py -s ../../03_HOX_WNT/wnt -g ../../02_REDUX_gene_density_analysis/genomes/ -c ../../01_microsynteny/chrom/ -m ../../07_wnt5_wnt7/NeuroWnt_density_manual_redone/manual_final.clusters -og /scratch/robert/2019_microsynteny_size_constraints/07_wnt5_wnt7/NeuroWnt_density_manual_redone/wnt5_7_graph_data/wnt5_wnt7.clus -o wnt

```


Regarding blocks, some of them a lineage specific expansions of CACNA1 or CACNA2D subunits. We filter the blocks for keeping only the ones with *wnt5* or *wnt7*