## 9 Graph representation of blocks
## 2. File preparation
get assembly length of the genomes. In the normali (different from lengths than the used for normalizing)
```
for i in $(ls 02_gene_density_analysis/genomes/*genome); do get_length_genome.py -n 0; done
```
Convert the blocks to edgelists (graphs where OGs are the nodes, and edges are the distances of syntenic orthogroups.
We also make edgelists for manually curated hox and wnt genes.
```
Block_to_OGcommus.py -c 02_gene_density_analysis/gene_density/second_sampling/Metazoa_total/Metazoa.m2.total.clusters -b 02_gene_density_analysis/gene_density/second_sampling/Metazoa_total/Metazoa.m2.total.synt -g ACAPL.chrom ACRMI.chrom ADIVA.chrom AMPQU.chrom ANOGA.chrom AURAU.chrom BRALA.chrom CAEEL.chrom CALMI.chrom CAPOW.chrom CAPTE.chrom CIOIN.chrom CLYHE.chrom CRAGI.chrom DAPPU.chrom DROME.chrom EUPSC.chrom EXAPA.chrom HELRO.chrom HOFMI.chrom HOIHO.chrom HYDVU.chrom IXOSC.chrom LATCH.chrom LINAN.chrom LOTGI.chrom MIZYE.chrom MNELE.chrom NEMVE.chrom PARTE.chrom PLEBA.chrom PTYFL.chrom SACKO.chrom SCHME.chrom STRMA.chrom STRPU.chrom SYCCI.chrom TRIAD.chrom TRICA.chrom XENTR.chrom -og 01_microsynteny/Orthofinder.clus -o 10_OG_graphs_representation/Metazoa_total -r 02_gene_density_analysis/gene_density/second_sampling/Metazoa_total/rand_blocks/Metazoa.m2.total.100r.synt

Block_to_OGcommus.py -c 02_gene_density_analysis/gene_density/second_sampling/Planu_placo/Planu_Placo.m2.novel.clusters -b 02_gene_density_analysis/gene_density/second_sampling/Planu_placo/Planu_Placo.m2.novel.synt -g ACRMI.chrom ADIVA.chrom ANOGA.chrom BRALA.chrom CAPTE.chrom CIOIN.chrom CRAGI.chrom EUPSC.chrom EXAPA.chrom HOIHO.chrom HYDVU.chrom LINAN.chrom LOTGI.chrom MIZYE.chrom PARTE.chrom PTYFL.chrom SACKO.chrom SCHME.chrom STRPU.chrom TRIAD.chrom TRICA.chrom -og 01_microsynteny/Orthofinder.clus -o 10_OG_graphs_representation/Planu_Placo_novel -r 02_gene_density_analysis/gene_density/second_sampling/Planu_placo/rand_blocks/Planu_Placo.m2.novel.100r.synt

Block_to_OGcommus.py -c 02_gene_density_analysis/gene_density/second_sampling/Bilateria/Bilateria.m2.novel.clusters -b 02_gene_density_analysis/gene_density/second_sampling/Bilateria/Bilateria.m2.novel.synt -g ACAPL.chrom ADIVA.chrom ANOGA.chrom BRALA.chrom CAEEL.chrom CALMI.chrom CAPTE.chrom CHEMY.chrom CIOIN.chrom CRAGI.chrom DANRE.chrom DAPPU.chrom DROME.chrom EUPSC.chrom GALGA.chrom HELRO.chrom HIPCO.chrom HOFMI.chrom HOMSA.chrom IXOSC.chrom LATCH.chrom LEPOC.chrom LINAN.chrom LOTGI.chrom MAYZE.chrom MIZYE.chrom MUSMU.chrom PARTE.chrom PTYFL.chrom SACKO.chrom SCHME.chrom STRMA.chrom STRPU.chrom TRICA.chrom XENTR.chrom -og 01_microsynteny/Orthofinder.clus -o 10_OG_graphs_representation/Bilateria_novel -r 02_gene_density_analysis/gene_density/second_sampling/Bilateria/rand_blocks/Bilateria.m2.novel.100r.synt

Block_to_OGcommus.py -c 02_gene_density_analysis/gene_density/second_sampling/Planulozoa/Planulozoa.m2.novel.clusters -b 02_gene_density_analysis/gene_density/second_sampling/Planulozoa/Planulozoa.m2.novel.synt -g ACAPL.chrom ACRMI.chrom ADIVA.chrom ANOGA.chrom AURAU.chrom BRALA.chrom CAEEL.chrom CALMI.chrom CAPTE.chrom CHEMY.chrom CIOIN.chrom CLYHE.chrom CRAGI.chrom DANRE.chrom DAPPU.chrom DROME.chrom EUPSC.chrom EXAPA.chrom GALGA.chrom HELRO.chrom HIPCO.chrom HOFMI.chrom HOMSA.chrom HYDVU.chrom IXOSC.chrom LATCH.chrom LEPOC.chrom LINAN.chrom LOTGI.chrom MAYZE.chrom MIZYE.chrom MUSMU.chrom NEMVE.chrom PARTE.chrom PTYFL.chrom SACKO.chrom SCHME.chrom STRMA.chrom STRPU.chrom TRICA.chrom XENTR.chrom -og 01_microsynteny/Orthofinder.clus -o 10_OG_graphs_representation/Planulozoa_novel -r 02_gene_density_analysis/gene_density/second_sampling/Planulozoa/rand_blocks/Planulozoa.m2.novel.100r.synt

Block_to_OGcommus.py -c 02_gene_density_analysis/gene_density/second_sampling/recent_blocks/Lophotrochozoa/Lophotrochozoa_novel.clusters -b 02_gene_density_analysis/gene_density/second_sampling/recent_blocks/Lophotrochozoa/Lophotrochozoa_novel.synt -g ADIVA.chrom CAPTE.chrom CRAGI.chrom EUPSC.chrom HELRO.chrom LINAN.chrom LOTGI.chrom MIZYE.chrom SCHME.chrom -og 01_microsynteny/Orthofinder.clus -o 10_OG_graphs_representation/Lophotrochozoa_novel -r 02_gene_density_analysis/gene_density/second_sampling/recent_blocks/Lophotrochozoa/rand_blocks/Lophotrochozoa_novel.100r.synt

Block_to_OGcommus.py -c 02_gene_density_analysis/gene_density/second_sampling/recent_blocks/Vertebrata/Vertebrata_novel.clusters -b 02_gene_density_analysis/gene_density/second_sampling/recent_blocks/Vertebrata/Vertebrata_novel.synt -g CALMI.chrom CHEMY.chrom DANRE.chrom GALGA.chrom HIPCO.chrom HOMSA.chrom LATCH.chrom LEPOC.chrom MAYZE.chrom MUSMU.chrom XENTR.chrom -og 01_microsynteny/Orthofinder.clus -o 10_OG_graphs_representation/Vertebrata_novel -r 02_gene_density_analysis/gene_density/second_sampling/recent_blocks/Vertebrata/rand_blocks/Vertebrata_novel.100r.synt


```

We use the lengths obtained with `get_length_genome.py` to normalise the basepairs distances in the OG pairs edgelists.


WNT graph
We'll use our annotation. There is 8 OGs that are annotated as being part of the syntenic block. We'll have 8 OGs in the 07_wnt5_wnt7/wnt5_7_graph_data/wnt5_wnt7.clus.

ATXN10, CACNA1, CACNA2,DCP1, ERC, FBXL14, NINJ, WNT
```
cd 01_microsynteny/chrom
Block_to_OGcommus_2.py -c 07_wnt5_wnt7/manual_final.clusters -b 07_wnt5_wnt7/manual_final.synt -g ACAPL.chrom ACRMI.chrom ANOGA.chrom BRALA.chrom CALMI.chrom CHEMY.chrom DANRE.chrom DAPPU.chrom EUPSC.chrom EXAPA.chrom GALGA.chrom HIPCO.chrom HOMSA.chrom IXOSC.chrom LATCH.chrom LEPOC.chrom LINAN.chrom LOTGI.chrom MAYZE.chrom MUSMU.chrom NEMVE.chrom PARTE.chrom SACKO.chrom TRICA.chrom XENTR.chrom  -og 07_wnt5_wnt7/wnt5_7_graph_data/wnt5_wnt7.clus -o 07_wnt5_wnt7/wnt5_7_graph_data/wnt5_wnt7 -r 07_wnt5_wnt7/rand_blocks/manual_final.100r.synt --custom_orthology
```

HOX graph
We'll use our annotation `03_HOX/hox_graph_data/hox_planu.clus`, where hoxes are grouped by taxonomic groups (Lophotrochozoan hox OGs, Vertebrate hox Ogs, etc.) specific.
```
cd 01_microsynteny/chrom
Block_to_OGcommus_2.py -c 03_HOX/HOX_density/5.blocks.3.syn.clusters -b 03_HOX/HOX_density/5.blocks.3.syn_corrected.synt -g ACAPL.chrom ACRMI.chrom ANOGA.chrom BRALA.chrom CALMI.chrom CAPTE.chrom CHEMY.chrom CIOIN.chrom CRAGI.chrom DAPPU.chrom DROME.chrom EUPSC.chrom EXAPA.chrom GALGA.chrom HELRO.chrom HIPCO.chrom HOMSA.chrom IXOSC.chrom LATCH.chrom LEPOC.chrom LINAN.chrom LOTGI.chrom MAYZE.chrom MIZYE.chrom MUSMU.chrom NEMVE.chrom PARTE.chrom PTYFL.chrom SACKO.chrom STRMA.chrom STRPU.chrom TRICA.chrom XENTR.chrom -og 03_HOX/hox_graph_data/hox_planu.clus -o 03_HOX/hox_graph_data/hox -r 03_HOX/HOX_density/rand_blocks/Hoxblocks.100r.synt --custom_orthology
```

